Upstream-Status: Backport

From 15cfff0eb7fb7ca58550f6259f0935ca7710a6b4 Mon Sep 17 00:00:00 2001
From: Stephen Street <stephen@redrocketcomputing.com>
Date: Tue, 5 Nov 2024 08:47:17 -0800
Subject: [PATCH] Do not leak build directory paths in sdk

Signed-off-by: Stephen Street <stephen@redrocketcomputing.com>
---
 PCLConfig.cmake.in | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/PCLConfig.cmake.in b/PCLConfig.cmake.in
index b6322c623..b7e290c23 100644
--- a/PCLConfig.cmake.in
+++ b/PCLConfig.cmake.in
@@ -23,6 +23,7 @@ if(POLICY CMP0074)
   # NEW behavior, we need to push a policy stack.
   cmake_policy(SET CMP0074 NEW)
 endif()
+cmake_policy(SET CMP0144 NEW)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/Modules")
 
@@ -91,9 +92,9 @@ macro(find_boost)
   if(PCL_ALL_IN_ONE_INSTALLER)
     set(BOOST_ROOT "${PCL_ROOT}/3rdParty/Boost")
   elseif(NOT BOOST_INCLUDEDIR)
-    set(BOOST_INCLUDEDIR "@Boost_INCLUDE_DIR@")
+    set(BOOST_INCLUDEDIR "/usr/include")
   endif()
-  
+
   set(Boost_ADDITIONAL_VERSIONS
     "@Boost_MAJOR_VERSION@.@Boost_MINOR_VERSION@.@Boost_SUBMINOR_VERSION@" "@Boost_MAJOR_VERSION@.@Boost_MINOR_VERSION@"
     "1.84.0" "1.84" "1.83.0" "1.83" "1.82.0" "1.82" "1.81.0" "1.81" "1.80.0" "1.80"
@@ -117,7 +118,7 @@ macro(find_eigen)
   if(PCL_ALL_IN_ONE_INSTALLER)
     set(EIGEN_ROOT "${PCL_ROOT}/3rdParty/Eigen")
   elseif(NOT EIGEN_ROOT)
-    get_filename_component(EIGEN_ROOT "@EIGEN_INCLUDE_DIRS@" ABSOLUTE)
+    get_filename_component(EIGEN_ROOT "/usr/include/eigen3" ABSOLUTE)
   endif()
   find_package(Eigen 3.3)
 endmacro()
@@ -225,7 +226,7 @@ macro(find_flann)
   if(PCL_ALL_IN_ONE_INSTALLER)
     set(FLANN_ROOT "${PCL_ROOT}/3rdParty/Flann")
   elseif(NOT FLANN_ROOT)
-    set(FLANN_ROOT "@FLANN_ROOT@")
+    set(FLANN_ROOT "/usr")
   endif()
 
   set(PCL_FLANN_REQUIRED_TYPE @PCL_FLANN_REQUIRED_TYPE@)
-- 
2.43.0

