From 1fdeab4413bcd0b365258c17868741cd61e7d368 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Wed, 4 Nov 2020 16:54:32 -0800
Subject: [PATCH] CMakeLists.txt: fetch contracts_lite with bitbake fetcher

Upstream-Status: Pending
---
 CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1584540..a5b7a36 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,9 +16,7 @@ set(contracts_lite_DESTDIR ${CMAKE_CURRENT_BINARY_DIR}/contracts_lite_install)
 
 include(ExternalProject)
 ExternalProject_Add(contracts_lite
-  GIT_REPOSITORY https://github.com/ros-safety/contracts_lite.git
-  GIT_TAG 0.4.0
-  GIT_SHALLOW TRUE
+  SOURCE_DIR ${CMAKE_SOURCE_DIR}/contracts-lite-upstream
   BUILD_IN_SOURCE TRUE
   INSTALL_DIR ${contracts_lite_DESTDIR}
   CONFIGURE_COMMAND ${CMAKE_COMMAND} -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR> -DCMAKE_POSITION_INDEPENDENT_CODE=ON .
