From 0971a8325ed0c69d7dd365b954a05366d4895e84 Mon Sep 17 00:00:00 2001
From: Windel Bouwman <windel@windel.nl>
Date: Thu, 11 Nov 2021 16:45:13 +0100
Subject: [PATCH] Remove optional pytest dependencies.

Upstream-Status: Pending

---
 setup.cfg | 4 ----
 1 file changed, 4 deletions(-)

Index: colcon-core-0.18.3/setup.cfg
===================================================================
--- colcon-core-0.18.3.orig/setup.cfg
+++ colcon-core-0.18.3/setup.cfg
@@ -32,10 +32,6 @@ install_requires =
 	EmPy
 	importlib-metadata; python_version < "3.8"
 	packaging
-	pytest
-	pytest-cov
-	pytest-repeat
-	pytest-rerunfailures
 	setuptools>=30.3.0
 	tomli>=1.0.0; python_version < "3.11"
 packages = find:
