From 224186a1af40cd3b7efb4f0648a3a4ca6ca1e7fa Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Thu, 6 Apr 2017 15:25:13 +0300
Subject: [PATCH] python.cmake: look for python3 first

Upstream-Status: Inappropriate [the patch overrides ROS's default to python2]

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
---
 cmake/python.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/python.cmake b/cmake/python.cmake
index 70f3d26..38f8320 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -5,6 +5,7 @@ if(PYTHON_VERSION)
   set(PythonInterp_FIND_VERSION "${PYTHON_VERSION}")
 endif()
 
+set(Python_ADDITIONAL_VERSIONS "3")
 find_package(PythonInterp REQUIRED)
 message(STATUS "Using PYTHON_EXECUTABLE: ${PYTHON_EXECUTABLE}")
 
-- 
2.9.3

