Remove Boost::system

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: moveit-planners-ompl-2.5.9-1/CMakeLists.txt
===================================================================
--- moveit-planners-ompl-2.5.9-1.orig/CMakeLists.txt
+++ moveit-planners-ompl-2.5.9-1/CMakeLists.txt
@@ -5,7 +5,7 @@ project(moveit_planners_ompl)
 find_package(moveit_common REQUIRED)
 moveit_package()
 
-find_package(Boost REQUIRED system filesystem date_time thread serialization)
+find_package(Boost REQUIRED filesystem date_time thread serialization)
 find_package(moveit_core REQUIRED)
 find_package(moveit_msgs REQUIRED)
 find_package(moveit_ros_planning REQUIRED)
