Use CMake Targets

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: moveit-planners-ompl-2.12.4-1/CMakeLists.txt
===================================================================
--- moveit-planners-ompl-2.12.4-1.orig/CMakeLists.txt
+++ moveit-planners-ompl-2.12.4-1/CMakeLists.txt
@@ -46,7 +46,7 @@ install(
   INCLUDES
   DESTINATION include/moveit_planners_ompl)
 ament_export_targets(moveit_planners_omplTargets HAS_LIBRARY_TARGET)
-ament_export_dependencies(moveit_core ompl)
+target_link_libraries(moveit_core::moveit_core INTERFACE ompl::ompl)

 pluginlib_export_plugin_description_file(moveit_core
                                          ompl_interface_plugin_description.xml)
