From 5a06a8525da6c0721f9a356ac631bc527b591f12 Mon Sep 17 00:00:00 2001
From: vmayoral <v.mayoralv@gmail.com>
Date: Fri, 9 Aug 2013 17:46:44 +0200
Subject: [PATCH] urdf CMakeLists fix

---
 urdf/CMakeLists.txt |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/urdf/CMakeLists.txt b/urdf/CMakeLists.txt
index 0afcb03..3448c05 100644
--- a/urdf/CMakeLists.txt
+++ b/urdf/CMakeLists.txt
@@ -29,8 +29,10 @@ if(APPLE)
   set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
 endif(APPLE)
 
-catkin_add_gtest(test_parser EXCLUDE_FROM_ALL test/test_robot_model_parser.cpp)
-target_link_libraries(test_parser ${PROJECT_NAME})
+if(CATKIN_ENABLE_TESTING)
+    catkin_add_gtest(test_parser EXCLUDE_FROM_ALL test/test_robot_model_parser.cpp)
+    target_link_libraries(test_parser ${PROJECT_NAME})
+endif()
 
 # no idea how CATKIN does this
 # rosbuild_add_rostest(${PROJECT_SOURCE_DIR}/test/test_robot_model_parser.launch)
-- 
1.7.9.5

