From 64d153e1cbd8a03657252505f510ae6b9de970ff Mon Sep 17 00:00:00 2001
From: Jan Vermaete <jan.vermaete@gmail.com>
Date: Mon, 10 Aug 2026 16:06:54 +0200
Subject: [PATCH 2/2] ptest: correct path of the to be checked python files

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Jan Vermaete <jan.vermaete@gmail.com>
---
 test/test_flake8.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/test_flake8.py b/test/test_flake8.py
index df62e99..db0f80b 100644
--- a/test/test_flake8.py
+++ b/test/test_flake8.py
@@ -3,6 +3,7 @@
 
 import logging
 from pathlib import Path
+import site
 import sys
 
 from flake8 import LOG
@@ -27,7 +28,7 @@ def test_flake8():
     sys.stdout = sys.stderr
     # implicitly calls report_errors()
     report = style_guide.check_files([
-        str(Path(__file__).parents[1] / 'colcon_bash'),
+        site.getsitepackages()[0] + '/' + 'colcon_bash',
     ])
     report_tests = style_guide_tests.check_files([
         str(Path(__file__).parents[1] / 'test'),
-- 
2.43.0

