diff -Naur a/cmake/genmsg-extras.cmake.em b/cmake/genmsg-extras.cmake.em
--- a/cmake/genmsg-extras.cmake.em	2013-01-19 20:01:11.000000000 +0100
+++ b/cmake/genmsg-extras.cmake.em	2013-03-13 16:39:51.000000000 +0100
@@ -172,8 +172,7 @@
     ${CATKIN_DEVEL_PREFIX}/share/${PROJECT_NAME}/cmake/${PROJECT_NAME}-msg-paths.cmake
     @@ONLY)
   # generate and install config of message include dirs for project
-  _prepend_path(${CMAKE_INSTALL_PREFIX}/share/${PROJECT_NAME} "${${PROJECT_NAME}_MSG_INCLUDE_DIRS_INSTALLSPACE}" INCLUDE_DIRS_W_PATH)
-  set(PKG_MSG_INCLUDE_DIRS "${INCLUDE_DIRS_W_PATH}")
+  set(PKG_MSG_INCLUDE_DIRS "${${PROJECT_NAME}_MSG_INCLUDE_DIRS_INSTALLSPACE}")
   configure_file(
     ${genmsg_CMAKE_DIR}/pkg-msg-paths.cmake.in
     ${CMAKE_CURRENT_BINARY_DIR}/catkin_generated/installspace/${PROJECT_NAME}-msg-paths.cmake
diff -Naur a/cmake/pkg-msg-paths.cmake.in b/cmake/pkg-msg-paths.cmake.in
--- a/cmake/pkg-msg-paths.cmake.in	2013-01-19 20:01:11.000000000 +0100
+++ b/cmake/pkg-msg-paths.cmake.in	2013-03-13 16:37:32.000000000 +0100
@@ -1,2 +1,5 @@
 set(@PROJECT_NAME@_MSG_INCLUDE_DIRS @PKG_MSG_INCLUDE_DIRS@)
+if(NOT IS_ABSOLUTE ${@PROJECT_NAME@_MSG_INCLUDE_DIRS})
+  set(@PROJECT_NAME@_MSG_INCLUDE_DIRS ${@PROJECT_NAME@_DIR}/../${@PROJECT_NAME@_MSG_INCLUDE_DIRS})
+endif()
 set(@PROJECT_NAME@_MSG_DEPENDENCIES @ARG_DEPENDENCIES@)
