From 044755e95bac8604da9e38cab9454770b60b158d Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 23 Apr 2020 09:29:33 -0700
Subject: [PATCH] CMakeLists.txt: fetch libyaml with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b467a07..9d5785b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,13 +61,7 @@ macro(build_libyaml)
   endif()
   include(ExternalProject)
   externalproject_add(libyaml-0.2.5
-    GIT_REPOSITORY https://github.com/yaml/libyaml.git
-    GIT_TAG 2c891fc7a770e8ba2fec34fc6b545c672beb37e6  # 0.2.5
-    GIT_CONFIG advice.detachedHead=false
-    # Suppress git update due to https://gitlab.kitware.com/cmake/cmake/-/issues/16419
-    # See https://github.com/ament/uncrustify_vendor/pull/22 for details
-    UPDATE_COMMAND ""
-    TIMEOUT 600
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/libyaml-upstream
     TIMEOUT 60
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libyaml_install
       -DINSTALL_INCLUDE_DIR:STRING=include/${PROJECT_NAME}
