From 13504e4674150cd305e2a7368fe315654f88c8e0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 23 Apr 2020 09:29:33 -0700
Subject: [PATCH] CMakeLists.txt: fetch libyaml with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54dae7b..86446c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,9 +61,7 @@ macro(build_libyaml)
   endif()
   include(ExternalProject)
   externalproject_add(libyaml-0.2.5
-    URL https://github.com/yaml/libyaml/archive/refs/tags/0.2.5.zip
-    URL_MD5 2464078985a73f9d298689d36061143f
-    TIMEOUT 60
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/libyaml-upstream
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libyaml_install
       ${extra_cmake_args}
