From e6f3979ff8b988fe7992bf08a8267f1e11f58627 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Tue, 4 Apr 2017 14:39:25 +0200
Subject: [PATCH] make rostest in CMakeLists optional (ros/rosdistro#3010)

Upstream-Status: Submitted [https://github.com/ros-planning/navigation/pull/565]

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 base_local_planner/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base_local_planner/CMakeLists.txt b/base_local_planner/CMakeLists.txt
index 2444ae1..507af18 100644
--- a/base_local_planner/CMakeLists.txt
+++ b/base_local_planner/CMakeLists.txt
@@ -10,7 +10,6 @@ find_package(catkin REQUIRED
             dynamic_reconfigure
             nav_core
             pcl_conversions
-            rostest
             costmap_2d
             pluginlib
             angles
@@ -129,6 +128,7 @@ install(DIRECTORY include/${PROJECT_NAME}/
 )
 
 if(CATKIN_ENABLE_TESTING)
+  find_package(rostest)
   catkin_add_gtest(base_local_planner_utest
     test/gtest_main.cpp
     test/utest.cpp
-- 
2.5.5

