From b59da4ce2e2bd38556abab192ab1a1240d50de39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADctor=20Mayoral=20Vilches?= <v.mayoralv@gmail.com>
Date: Tue, 10 May 2022 17:08:09 +0200
Subject: [PATCH] Deal with downstream recipe build errors
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Víctor Mayoral Vilches <v.mayoralv@gmail.com>
---
 cmake/rosidl_typesupport_c_generate_interfaces.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/rosidl_typesupport_c_generate_interfaces.cmake b/cmake/rosidl_typesupport_c_generate_interfaces.cmake
index 98eeed8..f79467c 100644
--- a/cmake/rosidl_typesupport_c_generate_interfaces.cmake
+++ b/cmake/rosidl_typesupport_c_generate_interfaces.cmake
@@ -47,7 +47,6 @@ endforeach()
 
 set(target_dependencies
   "${rosidl_typesupport_c_BIN}"
-  ${rosidl_typesupport_c_GENERATOR_FILES}
   "${rosidl_typesupport_c_TEMPLATE_DIR}/action__type_support.c.em"
   "${rosidl_typesupport_c_TEMPLATE_DIR}/idl__type_support.cpp.em"
   "${rosidl_typesupport_c_TEMPLATE_DIR}/msg__type_support.cpp.em"
-- 
2.34.1

