diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2790d6b9..b58fe895 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,7 +151,7 @@ install(
 
 install(
   TARGETS rmw_fastrtps_shared_cpp EXPORT rmw_fastrtps_shared_cpp
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION bin
 )
