--- old/Modules/FindBoost.cmake
+++ new/Modules/FindBoost.cmake
@@ -411,6 +411,22 @@
   set(${_ret} ${_boost_COMPILER} PARENT_SCOPE)
 endfunction()

+function(_Boost_consider_adding_pthreads _outvar)
+  # On Unix platforms (excluding cygwin) add pthreads to Boost_LIBRARIES
+  # if the user is searching for the boost-thread component.
+  if(UNIX AND NOT CYGWIN)
+    list(FIND Boost_FIND_COMPONENTS thread _using_boost_thread)
+    if(_using_boost_thread GREATER -1)
+      find_library(BOOST_THREAD_LIBRARY NAMES pthread
+                   DOC "The threading library used by boost-thread"
+      )
+      if(BOOST_THREAD_LIBRARY)
+        set(${_outvar} ${ARGN} ${BOOST_THREAD_LIBRARY} PARENT_SCOPE)
+      endif()
+    endif()
+  endif()
+endfunction()
+
 #
 # End functions/macros
 #
@@ -825,6 +841,7 @@
   list(APPEND _boost_LIBRARY_SEARCH_DIRS
     ${Boost_INCLUDE_DIR}/lib
     ${Boost_INCLUDE_DIR}/../lib
+    ${Boost_INCLUDE_DIR}/../lib/${CMAKE_LIBRARY_ARCHITECTURE}
     ${Boost_INCLUDE_DIR}/stage/lib
     )
   if( Boost_NO_SYSTEM_PATHS )
@@ -1106,6 +1123,9 @@
       list(APPEND Boost_LIBRARIES ${Boost_${UPPERCOMPONENT}_LIBRARY})
     endif()
   endforeach()
+
+  # Add pthread library on UNIX if thread component was found
+  _Boost_consider_adding_pthreads(Boost_LIBRARIES ${Boost_LIBRARIES})
 else()
   if(Boost_FIND_REQUIRED)
     message(SEND_ERROR "Unable to find the requested Boost libraries.\n${Boost_ERROR_REASON}")
