Use system-provided zenoh packages

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: zenoh-cpp-vendor-0.2.9-1/CMakeLists.txt
===================================================================
--- zenoh-cpp-vendor-0.2.9-1.orig/CMakeLists.txt
+++ zenoh-cpp-vendor-0.2.9-1/CMakeLists.txt
@@ -17,6 +17,8 @@ find_package(ament_cmake_vendor_package
 # latter is a list separater in cmake and hence the string will be split into two when expanded.
 set(ZENOHC_CARGO_FLAGS "--features=shared-memory zenoh/transport_serial")
 
+find_package(zenohc)
+
 # Set VCS_VERSION to 1.6.2 commits of zenoh/zenoh-c/zenoh-cpp to benefit from:
 #  * Various performance improvements:
 #    - https://github.com/eclipse-zenoh/zenoh/pull/2109
@@ -43,6 +45,7 @@ set(ZENOHC_CARGO_FLAGS "--features=share
 # * Precommit SHM pages, avoiding overcommit and possible crash if SHM exhausts
 #   - https://github.com/eclipse-zenoh/zenoh/pull/2175
 ament_vendor(zenoh_c_vendor
+  SATISFIED zenohc_FOUND
   VCS_URL https://github.com/eclipse-zenoh/zenoh-c.git
   VCS_VERSION f376456ccf75ed837a21a186bdf5191cba50eb3b
   CMAKE_ARGS
@@ -54,15 +57,15 @@ ament_vendor(zenoh_c_vendor
 
 ament_export_dependencies(zenohc)
 
+find_package(zenohcxx)
 ament_vendor(zenoh_cpp_vendor
+  SATISFIED zenohcxx_FOUND
   VCS_URL https://github.com/eclipse-zenoh/zenoh-cpp
   VCS_VERSION 0cd54f291039a65b96921a5951a66aeef088e67c
   CMAKE_ARGS
     -DZENOHCXX_ZENOHC=OFF
 )
 
-externalproject_add_stepdependencies(zenoh_cpp_vendor configure zenoh_c_vendor)
-
 ament_export_dependencies(zenohcxx)
 
 ament_package()
