Upstream-Status: Pending

From 3d13472f7489bda718ded20d885dbaf98e36442c Mon Sep 17 00:00:00 2001
From: Amini Allight <amini.allight@protonmail.com>
Date: Wed, 12 Feb 2025 04:48:13 +0000
Subject: [PATCH 3/7] updated docstrings

---
 websocketpp/transport/asio/endpoint.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/websocketpp/transport/asio/endpoint.hpp b/websocketpp/transport/asio/endpoint.hpp
index 13dda05..0545343 100644
--- a/websocketpp/transport/asio/endpoint.hpp
+++ b/websocketpp/transport/asio/endpoint.hpp
@@ -510,7 +510,7 @@ public:
      *
      * @param host A string identifying a location. May be a descriptive name or
      * a numeric address string.
-     * @param context A string identifying the requested service. This may be a
+     * @param service A string identifying the requested service. This may be a
      * descriptive name or a numeric string corresponding to a port number.
      * @param ec Set to indicate what error occurred, if any.
      */
@@ -628,7 +628,7 @@ public:
      *
      * @param host A string identifying a location. May be a descriptive name 
      * or a numeric address string.
-     * @param context A string identifying the requested service. This may be a
+     * @param service A string identifying the requested service. This may be a
      * descriptive name or a numeric string corresponding to a port number.
      * @param ec Set to indicate what error occurred, if any.
      */
-- 
2.43.0

