From 77497ae18875e3a1766448ac6b3758f896e2ddc8 Mon Sep 17 00:00:00 2001
From: "P. J. Reed" <preed@swri.org>
Date: Mon, 25 May 2020 11:34:33 -0500
Subject: [PATCH] Fix build with gpsd 3.20

Upstream-Status: Backport [https://github.com/swri-robotics/gps_umd/pull/28/commits/e0e9993816fe453b374b4e06b8ab53c5c9f56cb0]

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 src/client.cpp | 24 ++++++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/src/client.cpp b/src/client.cpp
index 443d5b4..ed5cca7 100644
--- a/src/client.cpp
+++ b/src/client.cpp
@@ -86,8 +86,13 @@ namespace gpsd_client
       if (p == nullptr)
         return;
 
-      if (!p->online)
+#if GPSD_API_MAJOR_VERSION >= 9
+      if (!p->online.tv_sec && !p->online.tv_nsec) {
+#else
+      if (!p->online) {
+#endif
         return;
+      }
 
       process_data_gps(p);
       process_data_navsat(p);
@@ -158,7 +163,11 @@ namespace gpsd_client
           status.status |= 18; // same here
 #endif
 
+#if GPSD_API_MAJOR_VERSION >= 9
+        fix.time = (double)(p->fix.time.tv_sec) + (double)(p->fix.time.tv_nsec) / 1000000.;
+#else
         fix.time = p->fix.time;
+#endif
         fix.latitude = p->fix.latitude;
         fix.longitude = p->fix.longitude;
         fix.altitude = p->fix.altitude;
@@ -209,10 +218,17 @@ namespace gpsd_client
 
       /* TODO: Support SBAS and other GBAS. */
 
-      if (use_gps_time_ && !std::isnan(p->fix.time))
+#if GPSD_API_MAJOR_VERSION >= 9
+      if (use_gps_time_ && (p->online.tv_sec || p->online.tv_nsec)) {
+        fix->header.stamp = rclcpp::Time(p->fix.time.tv_sec, p->fix.time.tv_nsec);
+#else
+      if (use_gps_time_ && !std::isnan(p->fix.time)) {
         fix->header.stamp = rclcpp::Time(p->fix.time);
-      else
+#endif
+      }
+      else {
         fix->header.stamp = this->get_clock()->now();
+      }
 
       fix->header.frame_id = frame_id_;
 
@@ -272,4 +288,4 @@ namespace gpsd_client
 }
 
 #include <rclcpp_components/register_node_macro.hpp>
-RCLCPP_COMPONENTS_REGISTER_NODE(gpsd_client::GPSDClientComponent)
\ No newline at end of file
+RCLCPP_COMPONENTS_REGISTER_NODE(gpsd_client::GPSDClientComponent)
