From: Jochen Sprickerhof <jochen@sprickerhof.de>
Date: Wed, 4 Jul 2012 17:38:49 +0200
Subject: Add SONAME to libraries

---
 Platform/Linux/Build/Common/CommonCppMakefile |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Platform/Linux/Build/Common/CommonCppMakefile b/Platform/Linux/Build/Common/CommonCppMakefile
index c2c4afc..d4f24c4 100644
--- a/Platform/Linux/Build/Common/CommonCppMakefile
+++ b/Platform/Linux/Build/Common/CommonCppMakefile
@@ -77,8 +77,8 @@ ifneq "$(LIB_NAME)" ""
 	CFLAGS += -fPIC -fvisibility=hidden
 	ifneq ("$(OSTYPE)","Darwin")
 		LDFLAGS += -Wl,--no-undefined
-		OUTPUT_NAME = lib$(LIB_NAME).so
-		OUTPUT_COMMAND = $(CXX) -o $(OUTPUT_FILE) $(OBJ_FILES) $(LDFLAGS) -shared
+		OUTPUT_NAME = lib$(LIB_NAME).so.0
+		OUTPUT_COMMAND = $(CXX) -o $(OUTPUT_FILE) $(OBJ_FILES) $(LDFLAGS) -shared -Wl,-soname,${OUTPUT_NAME} && cd ${OUT_DIR} && ln -fs ${OUTPUT_NAME} lib$(LIB_NAME).so
 	else
 		LDFLAGS += -undefined error
 		OUTPUT_NAME = lib$(LIB_NAME).dylib
