Index: git/Platform/Linux/Build/Makefile
===================================================================
--- git.orig/Platform/Linux/Build/Makefile
+++ git/Platform/Linux/Build/Makefile
@@ -29,16 +29,16 @@ ALL_UTILS = \
 #ALL_MONO_PROJS = \
 #	Wrappers/OpenNI.net
 
-ALL_JAVA_PROJS = \
-	Wrappers/OpenNI.jni \
-	Wrappers/OpenNI.java
+#ALL_JAVA_PROJS = \
+#	Wrappers/OpenNI.jni \
+#	Wrappers/OpenNI.java
 	
 # list all core projects
 ALL_CORE_PROJS = \
 	OpenNI \
 	$(ALL_MODULES) \
-	$(ALL_UTILS) \
-	$(ALL_JAVA_PROJS)
+	$(ALL_UTILS)
+#	$(ALL_JAVA_PROJS)
 
 # list all samples
 CORE_SAMPLES = \
@@ -74,14 +74,14 @@ endif
 #	Samples/SimpleViewer.net \
 #	Samples/UserTracker.net
 	
-JAVA_SAMPLES = \
-	Samples/SimpleRead.java \
-	Samples/SimpleViewer.java \
-	Samples/UserTracker.java \
+#JAVA_SAMPLES = \
+#	Samples/SimpleRead.java \
+#	Samples/SimpleViewer.java \
+#	Samples/UserTracker.java \
 
 ALL_SAMPLES = \
-	$(CORE_SAMPLES) \
-	$(JAVA_SAMPLES)
+	$(CORE_SAMPLES)
+#	$(JAVA_SAMPLES)
 
 # list all projects that are build
 ALL_BUILD_PROJS = \
