From ebc139766ccd7f82dfe48ca8951819b9f9e87c36 Mon Sep 17 00:00:00 2001
From: Steve Macenski <stevenmacenski@gmail.com>
Date: Sat, 9 May 2020 13:36:09 -0700
Subject: [PATCH] update for BT.cpp master (#1714)

Upstream-Status: Backport [https://github.com/ros-planning/navigation2/commit/50b044a863fa8b470bd862c72e1e5b30d672e840]
---
 src/ros_topic_logger.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ros_topic_logger.cpp b/src/ros_topic_logger.cpp
index e5c0c39e..278df1ee 100644
--- a/src/ros_topic_logger.cpp
+++ b/src/ros_topic_logger.cpp
@@ -21,7 +21,7 @@ namespace nav2_bt_navigator
 
 RosTopicLogger::RosTopicLogger(
   const rclcpp::Node::SharedPtr & ros_node, const BT::Tree & tree)
-: StatusChangeLogger(tree.root_node), ros_node_(ros_node)
+: StatusChangeLogger(tree.rootNode()), ros_node_(ros_node)
 {
   log_pub_ = ros_node_->create_publisher<nav2_msgs::msg::BehaviorTreeLog>(
     "behavior_tree_log",
