From 08b11b0481c4fb9f96d2df2dfca124ebea90620a Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Tue, 12 May 2020 11:52:56 -0700
Subject: [PATCH] CMakeLists.txt: don't use SYSTEM for ccd, eigen3 and octomap
 includes

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 src/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 64b81e5..7472b03 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -80,7 +80,7 @@ endif()
 if(TARGET ccd)
   target_link_libraries(${PROJECT_NAME} PUBLIC ccd)
 else()
-  target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC "${CCD_INCLUDE_DIRS}")
+  target_include_directories(${PROJECT_NAME} PUBLIC "${CCD_INCLUDE_DIRS}")
   target_link_libraries(${PROJECT_NAME} PUBLIC "${CCD_LIBRARIES}")
 endif()
 
@@ -93,9 +93,9 @@ if(TARGET Eigen3::Eigen)
   # linked
   target_link_libraries(${PROJECT_NAME} PUBLIC Eigen3::Eigen)
 elseif(EIGEN3_INCLUDE_DIRS)
-  target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC "${EIGEN3_INCLUDE_DIRS}")
+  target_include_directories(${PROJECT_NAME} PUBLIC "${EIGEN3_INCLUDE_DIRS}")
 else()
-  target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC "${EIGEN3_INCLUDE_DIR}")
+  target_include_directories(${PROJECT_NAME} PUBLIC "${EIGEN3_INCLUDE_DIR}")
 endif()
 
 if(FCL_HAVE_OCTOMAP)
@@ -104,7 +104,7 @@ if(FCL_HAVE_OCTOMAP)
   if(TARGET octomap)
     target_link_libraries(${PROJECT_NAME} PUBLIC octomap)
   elseif(OCTOMAP_INCLUDE_DIRS AND OCTOMAP_LIBRARIES)
-    target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC "${OCTOMAP_INCLUDE_DIRS}")
+    target_include_directories(${PROJECT_NAME} PUBLIC "${OCTOMAP_INCLUDE_DIRS}")
     target_link_libraries(${PROJECT_NAME} PUBLIC "${OCTOMAP_LIBRARIES}")
   endif()
 endif()
