Index: git/rc_genicam_api/buffer.cc
===================================================================
--- git.orig/rc_genicam_api/buffer.cc
+++ git/rc_genicam_api/buffer.cc
@@ -33,6 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <cstdint>
+
 #include "buffer.h"
 #include "stream.h"
 #include "config.h"
Index: git/rc_genicam_api/config.cc
===================================================================
--- git.orig/rc_genicam_api/config.cc
+++ git/rc_genicam_api/config.cc
@@ -33,6 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <cstdint>
+
 #include "config.h"
 #include "buffer.h"
 
Index: git/rc_genicam_api/image.cc
===================================================================
--- git.orig/rc_genicam_api/image.cc
+++ git/rc_genicam_api/image.cc
@@ -33,6 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <cstdint>
+
 #include "image.h"
 
 #include "exception.h"
Index: git/rc_genicam_api/imagelist.cc
===================================================================
--- git.orig/rc_genicam_api/imagelist.cc
+++ git/rc_genicam_api/imagelist.cc
@@ -33,6 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <cstdint>
+
 #include "imagelist.h"
 
 #include <algorithm>
Index: git/rc_genicam_api/image_store.cc
===================================================================
--- git.orig/rc_genicam_api/image_store.cc
+++ git/rc_genicam_api/image_store.cc
@@ -33,6 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <cstdint>
+
 #include "image_store.h"
 #include "pixel_formats.h"
 
Index: git/rc_genicam_api/pointcloud.cc
===================================================================
--- git.orig/rc_genicam_api/pointcloud.cc
+++ git/rc_genicam_api/pointcloud.cc
@@ -33,6 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <cstdint>
+
 #include "pointcloud.h"
 
 #include <iostream>
