From 9b63e8d0ef120b4512a2744a93fe967750f671bd Mon Sep 17 00:00:00 2001
From: Amelia Grace <agrace@bastiansolutions.com>
Date: Tue, 3 Mar 2026 13:12:15 -0700
Subject: [PATCH] remove Werror compile options

Signed-off-by: Amelia Grace <agrace@bastiansolutions.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a85f51c3..7999ca723 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.16)
 project(hardware_interface_testing LANGUAGES CXX)
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
-  add_compile_options(-Wall -Wextra -Werror=conversion -Werror=unused-but-set-variable -Werror=return-type -Werror=shadow)
+  add_compile_options(-Wall -Wextra)
 endif()
 
 set(THIS_PACKAGE_INCLUDE_DEPENDS
