From 43139304aba41046d6ce3ca49cd57cd280e1256f Mon Sep 17 00:00:00 2001
From: Jan Vermaete <jan.vermaete@oip.be>
Date: Sun, 21 Dec 2025 14:28:14 +0100
Subject: [PATCH] Use system-provided zenoh packages

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>
---
 CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ce9aa3..3fc2d1d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,8 @@ find_package(ament_cmake_vendor_package REQUIRED)
 # latter is a list separater in cmake and hence the string will be split into two when expanded.
 set(ZENOHC_CARGO_FLAGS "--features=shared-memory zenoh/transport_serial")
 
+find_package(zenohc)
+
 # Set VCS_VERSION to 1.6.2 commits of zenoh/zenoh-c/zenoh-cpp to benefit from:
 #  * Various performance improvements:
 #    - https://github.com/eclipse-zenoh/zenoh/pull/2109
@@ -45,6 +47,7 @@ set(ZENOHC_CARGO_FLAGS "--features=shared-memory zenoh/transport_serial")
 ament_vendor(zenoh_c_vendor
   VCS_URL https://github.com/eclipse-zenoh/zenoh-c.git
   VCS_VERSION f376456ccf75ed837a21a186bdf5191cba50eb3b
+  SATISFIED zenohc_FOUND
   CMAKE_ARGS
     "-DZENOHC_CARGO_FLAGS=${ZENOHC_CARGO_FLAGS}"
     "-DZENOHC_BUILD_WITH_UNSTABLE_API=TRUE"
@@ -54,15 +57,16 @@ ament_vendor(zenoh_c_vendor
 
 ament_export_dependencies(zenohc)
 
+find_package(zenohcxx)
+
 ament_vendor(zenoh_cpp_vendor
+  SATISFIED zenoh_FOUND
   VCS_URL https://github.com/eclipse-zenoh/zenoh-cpp
   VCS_VERSION 0cd54f291039a65b96921a5951a66aeef088e67c
   CMAKE_ARGS
     -DZENOHCXX_ZENOHC=OFF
 )
 
-externalproject_add_stepdependencies(zenoh_cpp_vendor configure zenoh_c_vendor)
-
 ament_export_dependencies(zenohcxx)
 
 ament_package()
-- 
2.43.0

