From f4ebe746cc8f0bed19c664189b826d4e332b8a34 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 18 Dec 2020 07:40:04 -0800
Subject: [PATCH] CMakeLists.txt: fetch tvm with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f3e5a9..4b0187a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,22 +33,14 @@ endif()
 include(ExternalProject)
 
 ExternalProject_Add(tvm-build
-  GIT_REPOSITORY https://github.com/apache/incubator-tvm
-  GIT_TAG v0.7.0
-  GIT_SHALLOW TRUE
-  BUILD_IN_SOURCE TRUE
-  PATCH_COMMAND git apply ${CMAKE_CURRENT_SOURCE_DIR}/config.cmake.patch
-  COMMAND cp <SOURCE_DIR>/cmake/config.cmake <SOURCE_DIR>
-  CONFIGURE_COMMAND ${CMAKE_COMMAND} -DCMAKE_POSITION_INDEPENDENT_CODE=ON
-  INSTALL_COMMAND ""
+  SOURCE_DIR ${PROJECT_SOURCE_DIR}/tvm-upstream
+  CONFIGURE_COMMAND ${CMAKE_COMMAND} -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_TOOLCHAIN_FILE=${PROJECT_SOURCE_DIR}/../toolchain.cmake
 )
 
 # Build DMLC
 ExternalProject_Add_Step(tvm-build dmlc-build
-  DEPENDEES patch
   WORKING_DIRECTORY <SOURCE_DIR>/3rdparty/dmlc-core
-  COMMAND ${CMAKE_COMMAND} --configure -DCMAKE_POSITION_INDEPENDENT_CODE=ON .
-  COMMAND ${CMAKE_MAKE_PROGRAM}
+  COMMAND ${CMAKE_COMMAND} --configure -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_TOOLCHAIN_FILE=${PROJECT_SOURCE_DIR}/../toolchain.cmake .
   BYPRODUCTS libdmlc.a
 )
 
