Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -68,40 +68,14 @@ include(GNUInstallDirs)
 
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/Modules ${CMAKE_MODULE_PATH})
 
-include(FetchContent)
-if(${CMAKE_VERSION} VERSION_LESS 3.14)
-  macro(FetchContent_MakeAvailable NAME)
-    FetchContent_GetProperties(${NAME})
-    if(NOT ${NAME}_POPULATED)
-      FetchContent_Populate(${NAME})
-      add_subdirectory(${${NAME}_SOURCE_DIR} ${${NAME}_BINARY_DIR})
-    endif(NOT ${NAME}_POPULATED)
-  endmacro(FetchContent_MakeAvailable)
-endif(${CMAKE_VERSION} VERSION_LESS 3.14)
-
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 find_package(nlohmann_json REQUIRED)
 find_package(tinyxml2 REQUIRED)
 
-FetchContent_Declare(steering_functions FETCHCONTENT_UPDATES_DISCONNECTED
-  GIT_REPOSITORY https://github.com/Fields2Cover/steering_functions.git
-  GIT_TAG 33fc010017efa1ef2c8a2d4779fcda94b4b30d20
-)
-FetchContent_Declare(spline FETCHCONTENT_UPDATES_DISCONNECTED
-  GIT_REPOSITORY https://github.com/Fields2Cover/spline.git
-  GIT_TAG 1b5d4bad29082997076b264de84ca6d46c2ae6ab
-)
-FetchContent_Declare(matplotlib_cpp FETCHCONTENT_UPDATES_DISCONNECTED
-  GIT_REPOSITORY https://github.com/Fields2Cover/matplotlib-cpp.git
-  GIT_TAG 75c15d0c907a4b68bca5ef97032302bd14ccab8e
-)
-
-FetchContent_MakeAvailable(
-  steering_functions
-  spline
-  matplotlib_cpp
-)
+add_subdirectory(steering_functions)
+add_subdirectory(spline)
+add_subdirectory(matplotlib-cpp)
 
 find_package(GDAL 3.0 REQUIRED)
 
