From fd17ddf74d1aa32fe48597f5097c62b076efb0b4 Mon Sep 17 00:00:00 2001
From: JeongBong Seo <jb.seo@lge.com>
Date: Wed, 10 Apr 2019 10:29:12 +0900
Subject: Pass toolchain.cmake to external project

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf46706..258a4e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,6 +51,7 @@ ExternalProject_Add(ros1_rosbag_storage
   URL_MD5 4c8b4c33165b223870f5d77bb697bef6
   TIMEOUT 60
   CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/rosbag_install -DCMAKE_NO_SYSTEM_FROM_IMPORTED=1
+  CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=${CMAKE_CURRENT_BINARY_DIR}/../toolchain.cmake
   PATCH_COMMAND
     ${PATCH_CMD} ${CMAKE_CURRENT_SOURCE_DIR}/resources/rosbag1_encryption_patch.diff &&
     ${PATCH_CMD} ${CMAKE_CURRENT_SOURCE_DIR}/resources/bagh.diff &&
-- 
2.17.1

