From 4d53e95b5f81871e51cec6f1b6f07a86e9b170cd Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Sun, 26 Apr 2020 12:29:13 -0700
Subject: [PATCH] CMakeLists.txt: fetch slic with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 22 +++++-----------------
 1 file changed, 5 insertions(+), 17 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 031f31ee1..567dd4301 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,24 +5,12 @@ include(ExternalProject)
 ## Find catkin macros and libraries
 ## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
 ## is used, also find other catkin packages
-# execute_process(COMMAND
-#   git config --global http.sslVerify false)
 ## System dependencies are found with CMake's conventions
 # find_package(Boost REQUIRED COMPONENTS system)
 ExternalProject_Add(
   slic
-  URL https://github.com/garaemon/SLIC-Superpixels/archive/78d9a2ba7ae1d3fee8c2ec774a52536c5f08f07c.zip
-  GIT_TAG 78d9a2ba7ae1d3fee8c2ec774a52536c5f08f07c
-  CMAKE_COMMAND echo "configure"
-  BUILD_COMMAND echo "build"
-  INSTALL_COMMAND echo "install"
-  )
-
-find_package( OpenCV REQUIRED )
-install(CODE "execute_process(COMMAND cmake -E make_directory ${CMAKE_INSTALL_PREFIX}/incldue/)")
-install(CODE "execute_process(COMMAND cmake -E copy slic.h ${CMAKE_INSTALL_PREFIX}/incldue/ WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/slic-prefix/src/slic/)")
-
-install(CODE "execute_process(COMMAND cmake -E make_directory ${CMAKE_INSTALL_PREFIX}/lib/)")
-install(CODE "execute_process(COMMAND gcc -I${OpenCV_INCLUDE_DIRS} -fPIC --shared -o ${CMAKE_INSTALL_PREFIX}/lib/libslic.so ${CMAKE_CURRENT_BINARY_DIR}/slic-prefix/src/slic/slic.cpp)")
-
-
+  SOURCE_DIR ${PROJECT_SOURCE_DIR}/slic-upstream
+  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${PROJECT_BINARY_DIR}/slic-install
+)
+install(DIRECTORY ${PROJECT_BINARY_DIR}/slic-install/include/ DESTINATION include)
+install(DIRECTORY ${PROJECT_BINARY_DIR}/slic-install/lib/ DESTINATION lib)
