--- festival.orig/src/modules/Lexicon/complex.cc	2004-09-29 08:56:56.000000000 +0000
+++ festival/src/modules/Lexicon/complex.cc	2020-02-25 17:26:02.513183325 +0000
@@ -64,7 +64,7 @@
     LIST_ent le2 = *((LIST_ent *)e2);
     int rcode;
 
-    if ((rcode = fcompare(le1->word,le2->word)) != 0)
+    if ((rcode = fcompare(le1->word,le2->word,NULL)) != 0)
 	return rcode;
     else if ((rcode = strcmp(le1->pos,le2->pos)) != 0)
 	return rcode;
