Upstream-Status: Backport

From f82cd350a32d38b4d46900f4a6acccc27c16e2f1 Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Tue, 27 Aug 2024 15:23:28 +0200
Subject: [PATCH] core: use NPY_API_SYMBOL_ATTRIBUTE for Numpy >= 2.x

---
 include/eigenpy/numpy.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/eigenpy/numpy.hpp b/include/eigenpy/numpy.hpp
index dcfd48dd..fd63f3b8 100644
--- a/include/eigenpy/numpy.hpp
+++ b/include/eigenpy/numpy.hpp
@@ -11,6 +11,11 @@
 #define PY_ARRAY_UNIQUE_SYMBOL EIGENPY_ARRAY_API
 #endif
 
+// For compatibility with Numpy 2.x
+// See
+// https://numpy.org/devdocs/reference/c-api/array.html#c.NPY_API_SYMBOL_ATTRIBUTE
+#define NPY_API_SYMBOL_ATTRIBUTE
+
 #include <numpy/numpyconfig.h>
 #ifdef NPY_1_8_API_VERSION
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
