From f8643ff5b923b8b8dc9d281ad6c5194c3b8eed70 Mon Sep 17 00:00:00 2001
From: Ralph Lange <ralph.lange@de.bosch.com>
Date: Sat, 3 Aug 2019 22:36:47 +0200
Subject: [PATCH] Changed build files for use of fmilibrary_vendor package.

Signed-off-by: Ralph Lange <ralph.lange@de.bosch.com>
---
 CMakeLists.txt | 20 ++++----------------
 package.xml    |  1 +
 2 files changed, 5 insertions(+), 16 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7eb694..0c3c140 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,19 +7,6 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
-include(ExternalProject)
-externalproject_add(FMILibraryProject
-  URL https://jmodelica.org/fmil/FMILibrary-2.0.3-src.zip
-  URL_MD5 53d8edd7442c31bcd3fb56477e574845
-  TIMEOUT 60
-)
-externalproject_get_property(FMILibraryProject INSTALL_DIR)
-set(FMILibraryProject_INCLUDE_DIR "${INSTALL_DIR}/src/install/include")
-set(FMILibraryProject_LIB_DIR "${INSTALL_DIR}/src/install/lib")
-
-add_library(${PROJECT_NAME}_fmilib STATIC IMPORTED)
-set_target_properties(${PROJECT_NAME}_fmilib PROPERTIES IMPORTED_LOCATION ${FMILibraryProject_LIB_DIR}/libfmilib.a)
-
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
@@ -31,12 +18,13 @@ find_package(rclcpp_components REQUIRED)
 find_package(rclcpp_lifecycle REQUIRED)
 find_package(rcutils REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(fmilibrary_vendor REQUIRED)
 
 add_library(${PROJECT_NAME} SHARED
   src/${PROJECT_NAME}/FMIAdapter.cpp
   src/${PROJECT_NAME}/FMIAdapterNode.cpp
 )
-add_dependencies(${PROJECT_NAME} FMILibraryProject)
+
 target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include> PRIVATE $<BUILD_INTERFACE:${FMILibraryProject_INCLUDE_DIR}>)
 ament_target_dependencies(${PROJECT_NAME}
   "rcl_interfaces"
@@ -44,8 +32,9 @@ ament_target_dependencies(${PROJECT_NAME}
   "rclcpp_components"
   "rclcpp_lifecycle"
   "std_msgs"
+  "fmilibrary_vendor"
 )
-target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_fmilib dl)
+target_link_libraries(${PROJECT_NAME} dl)
 install(DIRECTORY include/ DESTINATION include)
 install(TARGETS ${PROJECT_NAME} DESTINATION lib)
 rclcpp_components_register_nodes(${PROJECT_NAME} "fmi_adapter::FMIAdapterNode")
@@ -56,7 +45,6 @@ install(TARGETS ${PROJECT_NAME}_node DESTINATION lib/${PROJECT_NAME})
 
 install(DIRECTORY launch DESTINATION share/${PROJECT_NAME}/)
 
-
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
   ament_add_gtest(${PROJECT_NAME}_test
diff --git a/package.xml b/package.xml
index 022a6f8..41546cd 100644
--- a/package.xml
+++ b/package.xml
@@ -19,6 +19,7 @@
   <build_depend>rclcpp_components</build_depend>
   <build_depend>rclcpp_lifecycle</build_depend>
   <build_depend>std_msgs</build_depend>
+  <build_depend>fmilibrary_vendor</build_depend>
 
   <exec_depend>launch</exec_depend>
   <exec_depend>launch_ros</exec_depend>
