Index: Source/Utilities.h
===================================================================
diff --git a/Source/Utilities.h b/Source/Utilities.h
--- a/Source/Utilities.h	(revision 1829)
+++ b/Source/Utilities.h	(revision 1830)
@@ -529,7 +529,8 @@
 static const char *FI_MSG_ERROR_DIB_MEMORY = "DIB allocation failed, maybe caused by an invalid image size or by a lack of memory";
 static const char *FI_MSG_ERROR_PARSING = "Parsing error";
 static const char *FI_MSG_ERROR_MAGIC_NUMBER = "Invalid magic number";
-static const char *FI_MSG_ERROR_UNSUPPORTED_FORMAT = "Unsupported format";
+static const char *FI_MSG_ERROR_UNSUPPORTED_FORMAT = "Unsupported image format";
+static const char *FI_MSG_ERROR_INVALID_FORMAT = "Invalid file format";
 static const char *FI_MSG_ERROR_UNSUPPORTED_COMPRESSION = "Unsupported compression type";
 static const char *FI_MSG_WARNING_INVALID_THUMBNAIL = "Warning: attached thumbnail cannot be written to output file (invalid format) - Thumbnail saving aborted";
 
