--- log4cxx-0.9.7/src/socketoutputstream.cpp~	2009-05-21 10:06:47.000000000 +0100
+++ log4cxx-0.9.7/src/socketoutputstream.cpp	2009-05-21 10:06:47.000000000 +0100
@@ -13,7 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
- 
+
+#include <string.h> 
 #include <log4cxx/helpers/socketoutputstream.h>
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/loglog.h>

--- log4cxx-0.9.7/src/stringtokenizer.cpp~	2009-05-21 10:11:19.000000000 +0100
+++ log4cxx-0.9.7/src/stringtokenizer.cpp	2009-05-21 10:11:20.000000000 +0100
@@ -13,7 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
- 
+
+#include <string.h> 
 #include <log4cxx/helpers/stringtokenizer.h>
 
 using namespace log4cxx;

--- log4cxx-0.9.7/src/timezone.cpp~	2009-05-21 10:14:27.000000000 +0100
+++ log4cxx-0.9.7/src/timezone.cpp	2009-05-21 10:14:27.000000000 +0100
@@ -13,7 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
- 
+
+#include <string.h>
 #include <log4cxx/helpers/timezone.h>
 #include <locale>
 
--- log4cxx-0.9.7/src/xmlsocketappender.cpp~	2009-05-21 10:16:24.000000000 +0100
+++ log4cxx-0.9.7/src/xmlsocketappender.cpp	2009-05-21 10:16:25.000000000 +0100
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <string.h>
 #include <log4cxx/net/xmlsocketappender.h>
 #include <log4cxx/helpers/loglog.h>
 #include <log4cxx/helpers/socketoutputstream.h>
