--- apache-log4cxx-0.10.0.orig//src/examples/cpp/console.cpp	2008-04-01 00:34:52.000000000 +0200
+++ apache-log4cxx-0.10.0//src/examples/cpp/console.cpp	2010-05-04 23:52:38.270096842 +0200
@@ -23,6 +23,8 @@
 #include <iostream>
 #include <locale.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
--- apache-log4cxx-0.10.0.orig//src/main/cpp/inputstreamreader.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0//src/main/cpp/inputstreamreader.cpp	2010-05-04 23:45:21.583105063 +0200
@@ -21,6 +21,8 @@
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
--- apache-log4cxx-0.10.0.orig//src/main/cpp/socketoutputstream.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0//src/main/cpp/socketoutputstream.cpp	2010-05-04 23:49:22.838093204 +0200
@@ -20,6 +20,8 @@
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
