diff -Naur a/cmake/genpy-extras.cmake.em b/cmake/genpy-extras.cmake.em
--- a/cmake/genpy-extras.cmake.em	2012-12-22 03:39:57.000000000 +0100
+++ b/cmake/genpy-extras.cmake.em	2013-05-22 10:46:08.000000000 +0200
@@ -1,11 +1,16 @@
 @[if DEVELSPACE]@
 # bin dir variables in develspace
 set(GENPY_BIN_DIR @(CMAKE_CURRENT_SOURCE_DIR)/scripts)
+set(CATKIN_FIND_ROOT_PATH_MODE NO_CMAKE_FIND_ROOT_PATH)
 @[else]@
 # bin dir variables in installspace
 set(GENPY_BIN_DIR @(CMAKE_INSTALL_PREFIX)/@(CATKIN_PACKAGE_BIN_DESTINATION))
+set(CATKIN_FIND_ROOT_PATH_MODE "")
 @[end if]@
 
+find_program(GENMSG_PY_BIN genmsg_py.py PATHS ${GENPY_BIN_DIR} NO_DEFAULT_PATH CATKIN_FIND_ROOT_PATH_MODE)
+find_program(GENSRV_PY_BIN gensrv_py.py PATHS ${GENPY_BIN_DIR} NO_DEFAULT_PATH CATKIN_FIND_ROOT_PATH_MODE)
+
 set(GENMSG_PY_BIN ${GENPY_BIN_DIR}/genmsg_py.py)
 set(GENSRV_PY_BIN ${GENPY_BIN_DIR}/gensrv_py.py)
 
