From f0cdc4c78b472ef1c9e02960111531275b51b9e5 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Wed, 4 Nov 2020 16:54:32 -0800
Subject: [PATCH] CMakeLists.txt: fetch mimick with bitbake fetcher

Upstream-Status: Pending
---
 CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c19c86..3113c06 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,12 +59,10 @@ macro(build_mimick)
 
   include(ExternalProject)
   externalproject_add(mimick-ext
-    GIT_REPOSITORY https://github.com/ros2/Mimick.git
-    GIT_TAG 99a35f3d2067708931945c64ac9caee80a0ef50e
-    TIMEOUT 6000
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/mimick-upstream
     ${cmake_commands}
     CMAKE_ARGS
-      ${cmake_configure_args}
+      ${cmake_configure_args} -DCMAKE_TOOLCHAIN_FILE=${PROJECT_SOURCE_DIR}/../toolchain.cmake
   )
 
   # The external project will install to the build folder, but we'll install that on make install.
