diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca0f133..fc2b0d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,20 +44,12 @@ macro(build_uncrustify)
 
   # Get uncrustify 0.72.0
   ExternalProject_Add(uncrustify-0.72.0
-    GIT_REPOSITORY https://github.com/uncrustify/uncrustify.git
-    GIT_TAG 1d3d8fa5e81bece0fac4b81316b0844f7cc35926  # uncrustify-0.72.0
-    GIT_CONFIG advice.detachedHead=false
-    # Suppress git update due to https://gitlab.kitware.com/cmake/cmake/-/issues/16419
-    # See https://github.com/ament/uncrustify_vendor/pull/22 for details
-    UPDATE_COMMAND ""
-    TIMEOUT 600
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/uncrustify-upstream
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_install
       ${extra_cmake_args}
       -Wno-dev
       --no-warn-unused-cli
-    PATCH_COMMAND
-      ${Patch_EXECUTABLE} -p1 -N < ${CMAKE_CURRENT_SOURCE_DIR}/install-patch.diff
   )
 
   # The external project will install to the build folder, but we'll install that on make install.
