Do not install documentation

Upstream-Status: Backport [https://raw.githubusercontent.com/ros2-gbp/uncrustify_vendor-release/e2bddd9df7abe1099ea9788e5b3811adabe9481e/install-patch.diff]

diff --git a/CMakeLists.txt b/CMakeLists.txt
index db0a6bfb..26ccf1a5 100644
--- a/CMakeLists.txt	2021-12-10 16:10:01.557816597 -0500
+++ b/CMakeLists.txt	2021-12-10 16:11:16.521464522 -0500
@@ -500,32 +500,33 @@
 # Install
 #
 if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-  install(TARGETS uncrustify DESTINATION ".")
-  install(FILES ${uncrustify_docs}
-    DESTINATION "."
-  )
-  install(FILES "${PROJECT_SOURCE_DIR}/documentation/htdocs/index.html"
-    DESTINATION "doc"
-  )
-  install(DIRECTORY "${PROJECT_SOURCE_DIR}/etc/"
-    DESTINATION "cfg"
-    FILES_MATCHING PATTERN "*.cfg"
-  )
+  install(TARGETS uncrustify RUNTIME DESTINATION bin)
+  # install(TARGETS uncrustify DESTINATION ".")
+  # install(FILES ${uncrustify_docs}
+  #   DESTINATION "."
+  # )
+  # install(FILES "${PROJECT_SOURCE_DIR}/documentation/htdocs/index.html"
+  #   DESTINATION "doc"
+  # )
+  # install(DIRECTORY "${PROJECT_SOURCE_DIR}/etc/"
+  #   DESTINATION "cfg"
+  #   FILES_MATCHING PATTERN "*.cfg"
+  # )
 else()
   include(GNUInstallDirs)
   install(TARGETS uncrustify
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   )
-  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/uncrustify.1"
-    DESTINATION "${CMAKE_INSTALL_MANDIR}/man1"
-  )
-  install(DIRECTORY "${PROJECT_SOURCE_DIR}/etc/"
-    DESTINATION "${CMAKE_INSTALL_DOCDIR}/examples"
-    FILES_MATCHING PATTERN "*.cfg"
-  )
-  install(FILES ${uncrustify_docs}
-    DESTINATION "${CMAKE_INSTALL_DOCDIR}"
-  )
+  # install(FILES "${CMAKE_CURRENT_BINARY_DIR}/uncrustify.1"
+  #   DESTINATION "${CMAKE_INSTALL_MANDIR}/man1"
+  # )
+  # install(DIRECTORY "${PROJECT_SOURCE_DIR}/etc/"
+  #   DESTINATION "${CMAKE_INSTALL_DOCDIR}/examples"
+  #   FILES_MATCHING PATTERN "*.cfg"
+  # )
+  # install(FILES ${uncrustify_docs}
+  #   DESTINATION "${CMAKE_INSTALL_DOCDIR}"
+  # )
 endif()
 
 #
