From 4c1945fb38a39b3a139d043dc21cb745f313bcc1 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Sun, 13 Oct 2013 17:41:58 +0200
Subject: [PATCH] check for CATKIN_ENABLE_TESTING

---
 CMakeLists.txt | 6 ++++--
 package.xml    | 2 +-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cee35b..a628238 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,8 +20,10 @@ include_directories(include ${catkin_INCLUDE_DIRS})
 add_library(laser_geometry src/laser_geometry.cpp)
 target_link_libraries(laser_geometry ${Boost_LIBRARIES} ${tf_LIBRARIES})
 
-catkin_add_gtest(projection_test test/projection_test.cpp)
-target_link_libraries(projection_test laser_geometry)
+if(CATKIN_ENABLE_TESTING)
+  catkin_add_gtest(projection_test test/projection_test.cpp)
+  target_link_libraries(projection_test laser_geometry)
+endif()
 
 install(TARGETS laser_geometry
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
diff --git a/package.xml b/package.xml
index 83e02e2..b000bde 100644
--- a/package.xml
+++ b/package.xml
@@ -15,7 +15,7 @@
 
   <url>http://ros.org/wiki/laser_geometry</url>
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend version_gte="0.5.68">catkin</buildtool_depend>
 
   <build_depend>sensor_msgs</build_depend>
   <build_depend>roscpp</build_depend>
-- 
1.8.1.2

