From f066ff531ecdea3b859881aa1f4b4b34cba910c1 Mon Sep 17 00:00:00 2001
From: Dirk Thomas <dthomas@osrfoundation.org>
Date: Wed, 3 Jul 2013 11:30:06 -0700
Subject: [PATCH] check for CATKIN_ENABLE_TESTING

---
 CMakeLists.txt | 4 +++-
 package.xml    | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a51426..20421f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,4 +32,6 @@ install(DIRECTORY include/${PROJECT_NAME}/
   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
   FILES_MATCHING PATTERN "*.h")
 
-add_subdirectory(test)
+if(CATKIN_ENABLE_TESTING)
+  add_subdirectory(test)
+endif()
diff --git a/package.xml b/package.xml
index 4d692ee..e45d984 100644
--- a/package.xml
+++ b/package.xml
@@ -14,7 +14,7 @@
   <author>Eitan Marder-Eppstein</author>
   <author>Vijay Pradeep</author>
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend version_gte="0.5.68">catkin</buildtool_depend>
 
   <build_depend>actionlib_msgs</build_depend>
   <build_depend>boost</build_depend>
-- 
1.8.1.6

