From 313ec3dcc07a04209d0359d62910e3a6394b57cb Mon Sep 17 00:00:00 2001
From: Stephen Street <stephen@redrocketcomputing.com>
Date: Fri, 8 Nov 2024 15:52:59 -0800
Subject: [PATCH] CMakeLists.txt: prevent building zstd with ExternalProject

Signed-off-by: Stephen Street <stephen@redrocketcomputing.com>
---
 CMakeLists.txt | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ec3c07a1..d205c812f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,19 +8,14 @@ list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 # We need at least VERSION 1.5.5, version check is done in Findzstd.cmake
 # The platform with the oldest supported version is RHEL-9,
 # which has zstd 1.5.1.  Make sure we have at least that version.
-find_package(zstd 1.5.1 QUIET)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(ZSTD REQUIRED libzstd>=1.5.5)
 
-ament_vendor(zstd_vendor
-  SATISFIED ${zstd_FOUND}
-  VCS_URL https://github.com/facebook/zstd.git
-  VCS_VERSION v1.5.5
-  SOURCE_SUBDIR build/cmake
-  CMAKE_ARGS
-    -DZSTD_BUILD_STATIC:BOOL=OFF
-    -DZSTD_BUILD_SHARED:BOOL=ON
-    -DZSTD_BUILD_PROGRAMS:BOOL=OFF
-    PATCHES patches
-)
+if(NOT ZSTD_FOUND)
+  message(STATUS "Zstd not found, missing dependency or version less than 1.5.5, found ${ZSTD_VERSION}")
+else()
+  message(STATUS "Found Zstd, skipping build.")
+endif()
 
 install(DIRECTORY cmake DESTINATION share/${PROJECT_NAME})
 
