From e38125ec4124306b3d1b6db882fb245b804c0b25 Mon Sep 17 00:00:00 2001
From: Mark Moll <mmoll@rice.edu>
Date: Mon, 26 Aug 2019 14:50:11 -0500
Subject: [PATCH] compilation fix for boost 1.71

---
 demos/PlannerData.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/demos/PlannerData.cpp b/demos/PlannerData.cpp
index 109338b..8802b7e 100644
--- a/demos/PlannerData.cpp
+++ b/demos/PlannerData.cpp
@@ -170,13 +170,15 @@ void readPlannerData()
         ob::GoalState goal(si);
         goal.setState(data.getGoalVertex(0).getState());
         ob::PlannerData::Graph::Vertex start = boost::vertex(data.getStartIndex(0), graph);
+        boost::astar_visitor<boost::null_visitor> dummy_visitor;
         boost::astar_search(graph, start,
             [&goal, &opt, &vertices](ob::PlannerData::Graph::Vertex v1) { return distanceHeuristic(v1, &goal, &opt, vertices); },
             boost::predecessor_map(prev).
             distance_compare([&opt](ob::Cost c1, ob::Cost c2) { return opt.isCostBetterThan(c1, c2); }).
             distance_combine([&opt](ob::Cost c1, ob::Cost c2) { return opt.combineCosts(c1, c2); }).
             distance_inf(opt.infiniteCost()).
-            distance_zero(opt.identityCost()));
+            distance_zero(opt.identityCost()).
+            visitor(dummy_visitor));
 
         // Extracting the path
         og::PathGeometric path(si);
