From 47ec5114fe70f25cbf658a7e9d755009c3574024 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Sun, 26 Apr 2020 12:29:13 -0700
Subject: [PATCH] CMakeLists.txt: fetch libcmt with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c98d37d6..aa4e4c2f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,15 +5,13 @@ include(ExternalProject)
 ## Find catkin macros and libraries
 ## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
 ## is used, also find other catkin packages
-# execute_process(COMMAND
-#   git config --global http.sslVerify false)
 ## System dependencies are found with CMake's conventions
 # find_package(Boost REQUIRED COMPONENTS system)
 ExternalProject_Add(
   libcmt
-  URL https://github.com/delmottea/libCMT/archive/e4d7ea42edafe13b1070ef4d595b2d6062d79d1a.zip
+  SOURCE_DIR ${PROJECT_SOURCE_DIR}/libcmt-upstream
   PATCH_COMMAND patch -p1 < ${PROJECT_SOURCE_DIR}/fix_opencv3.patch
-  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX} -DCMAKE_TOOLCHAIN_FILE=${PROJECT_SOURCE_DIR}/../toolchain.cmake -DCMAKE_NO_SYSTEM_FROM_IMPORTED=ON
   INSTALL_COMMAND echo "install"
   )
 
