Upstream-Status: Backport [https://github.com/moveit/random_numbers/commit/2a96c45380c01818b28ece17d432319ced7a33d8]

From 2a96c45380c01818b28ece17d432319ced7a33d8 Mon Sep 17 00:00:00 2001
From: Robert Haschke <rhaschke@techfak.uni-bielefeld.de>
Date: Sun, 1 Jun 2025 19:36:18 +0200
Subject: [PATCH] Fix declaration of include directories

---
 CMakeLists.txt | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99a8807..296a6db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,12 +4,14 @@ project(random_numbers)
 find_package(ament_cmake REQUIRED)
 find_package(Boost REQUIRED date_time random thread)

-include_directories(include)
-
 add_library(${PROJECT_NAME} SHARED
   src/random_numbers.cpp
 )
-target_link_libraries(${PROJECT_NAME} Boost::random)
+target_include_directories(${PROJECT_NAME} PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
+)
+target_link_libraries(${PROJECT_NAME} PUBLIC Boost::random)
 ament_export_targets(export_${PROJECT_NAME})
 ament_export_libraries(${PROJECT_NAME})
 ament_export_dependencies(Boost)
@@ -25,10 +27,7 @@ install(
   RUNTIME DESTINATION bin
 )

-install(
-  DIRECTORY include/
-  DESTINATION include
-)
+install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})

 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
