Upstream-Status: Backport [https://github.com/moveit/random_numbers/commit/501ebba9e49436cb003c308d03ad998ba4818ec7]

From 501ebba9e49436cb003c308d03ad998ba4818ec7 Mon Sep 17 00:00:00 2001
From: "David V. Lu!!" <davidvlu@gmail.com>
Date: Tue, 27 May 2025 14:38:38 -0400
Subject: [PATCH] Replace deprecated ament_target_libraries with
 target_link_libraries (#47)

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d87b506..e0d6fc6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,8 +14,8 @@ include_directories(include)
 add_library(${PROJECT_NAME} SHARED
   src/random_numbers.cpp
 )
-
-ament_target_dependencies(${PROJECT_NAME} Boost)
+target_link_libraries(${PROJECT_NAME} Boost::random)
+ament_export_targets(export_${PROJECT_NAME})
 ament_export_libraries(${PROJECT_NAME})
 ament_export_dependencies(Boost)
 ament_export_include_directories(include)
@@ -24,6 +24,7 @@ set_target_properties(${PROJECT_NAME} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE

 install(
   TARGETS ${PROJECT_NAME}
+  EXPORT export_${PROJECT_NAME}
   ARCHIVE DESTINATION lib
   LIBRARY DESTINATION lib
   RUNTIME DESTINATION bin
