diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2bfdbab7..4be2f2fa3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,11 +1,6 @@
 cmake_minimum_required(VERSION 3.16)
 project(steering_controllers_library LANGUAGES CXX)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror=conversion -Werror=unused-but-set-variable
-                      -Werror=return-type -Werror=shadow -Werror=format)
-endif()
-
 # find dependencies
 set(THIS_PACKAGE_INCLUDE_DEPENDS
   control_msgs
