diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b9cda78..206d55a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -145,15 +145,15 @@ endif()
 
 install(
   TARGETS component component_manager EXPORT export_${PROJECT_NAME}
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION bin
 )
 
 # Install executables
 install(
   TARGETS component_container component_container_mt component_container_isolated
-  RUNTIME DESTINATION lib/${PROJECT_NAME}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}
 )
 
 # Install include directories
