Upstream-Status: Pending

--- a/CMakeLists.txt	2018-05-31 11:49:33.000000000 -0700
+++ b/CMakeLists.txt	2019-08-16 15:43:11.106602301 -0700
@@ -17,6 +17,7 @@
   "1.ros_distro"
   "1.ros_version"
   "1.ros_python_version"
+  "1.ros_os_override"
 )
 if(CMAKE_HOST_UNIX)
   set(shell "sh")
--- /dev/null	2019-06-20 17:41:34.556355144 -0700
+++ b/env-hooks/1.ros_os_override.sh.in	2019-08-16 15:45:12.266604143 -0700
@@ -0,0 +1,3 @@
+# generated from ros_environment/env-hooks/1.ros_os_override.sh.in
+
+export ROS_OS_OVERRIDE="openembedded"
