From 3eaf40260ea400cb779ca229cbf3ab27f5a2b6eb Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Sat, 21 Sep 2013 16:33:11 +0200
Subject: [PATCH] rostest is optional

---
 rospy_tutorials/CMakeLists.txt |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/rospy_tutorials/CMakeLists.txt b/rospy_tutorials/CMakeLists.txt
index b588dfc..a48ec0d 100644
--- a/rospy_tutorials/CMakeLists.txt
+++ b/rospy_tutorials/CMakeLists.txt
@@ -1,7 +1,11 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(rospy_tutorials)
 
-find_package(catkin REQUIRED COMPONENTS message_generation rostest std_msgs)
+if(CATKIN_ENABLE_TESTING)
+  find_package(catkin REQUIRED COMPONENTS message_generation rostest std_msgs)
+else()
+  find_package(catkin REQUIRED COMPONENTS message_generation std_msgs)
+endif()
 
 add_message_files(DIRECTORY msg FILES Floats.msg HeaderString.msg)
 add_service_files(DIRECTORY srv FILES AddTwoInts.srv BadTwoInts.srv)
-- 
1.7.10.4

