From b037b68273c4557990afbaf44aa2413fb0c9cc57 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 23 Apr 2020 09:29:33 -0700
Subject: [PATCH] CMakeLists.txt: fetch libyaml with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99e6826..7f5433d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,8 +46,7 @@ macro(build_libyaml)
   endif()
   include(ExternalProject)
   ExternalProject_Add(libyaml-10c9078
-    URL https://github.com/yaml/libyaml/archive/10c907871f1ccd779c7fccf6b81a62762b5c4e7b.zip
-    URL_MD5 b595f0ab0735c04c7f6ed0a798a3eff7
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/libyaml-upstream
     TIMEOUT 60
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libyaml_install
