From 865299efcd5c34a6ae57d8751f9e74d0009ee5b6 Mon Sep 17 00:00:00 2001
From: Stephen Street <stephen@redrocketcomputing.com>
Date: Wed, 5 Feb 2025 16:00:57 -0800
Subject: [PATCH] Disable compiler options

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Stephen Street <stephen@redrocketcomputing.com>
---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8bbc3e..31a722c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,11 +1,6 @@
 cmake_minimum_required(VERSION 3.16)
 project(control_toolbox LANGUAGES CXX)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror=conversion -Werror=unused-but-set-variable
-                      -Werror=return-type -Werror=shadow -Werror=format)
-endif()
-
 if(WIN32)
   add_compile_definitions(
     # For math constants
-- 
2.43.0

