From d8ce655d595b4b44862399ed3781ebe1458b81a0 Mon Sep 17 00:00:00 2001
From: toship-botsync <toship.s@botsync.co>
Date: Tue, 15 Dec 2020 14:39:39 +0530
Subject: [PATCH] Merged volta_development.

Upstream-Status: Backport [https://github.com/botsync/volta/commit/57ecf986699ef7fac8345938b30e9856313f0ea9]
Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb55304..1815809 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,3 +7,15 @@ find_package(catkin REQUIRED COMPONENTS
   joint_state_controller
   twist_mux
 )
+
+catkin_package(
+)
+
+include_directories(
+        ${catkin_INCLUDE_DIRS}
+)
+
+install(
+        DIRECTORY launch config
+        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
