Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -77,12 +77,10 @@ if(${CMAKE_VERSION} VERSION_LESS 3.14)
 endif(${CMAKE_VERSION} VERSION_LESS 3.14)
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-FetchContent_Declare(json FETCHCONTENT_UPDATES_DISCONNECTED
-  URL https://github.com/nlohmann/json/releases/download/v3.10.5/json.tar.xz
-)
-FetchContent_Declare(tinyxml2 FETCHCONTENT_UPDATES_DISCONNECTED
-  URL https://github.com/leethomason/tinyxml2/archive/refs/tags/9.0.0.tar.gz
-)
+
+find_package(nlohmann_json REQUIRED)
+find_package(tinyxml2 REQUIRED)
+
 FetchContent_Declare(steering_functions FETCHCONTENT_UPDATES_DISCONNECTED
   GIT_REPOSITORY https://github.com/Fields2Cover/steering_functions.git
 )
@@ -94,8 +92,6 @@ FetchContent_Declare(matplotlib_cpp FETC
 )
 
 FetchContent_MakeAvailable(
-  json
-  tinyxml2
   steering_functions
   spline
   matplotlib_cpp
@@ -214,9 +210,7 @@ set_target_properties(Fields2Cover
 install(
   TARGETS
     Fields2Cover
-    tinyxml2
     steering_functions
-    nlohmann_json
   EXPORT Fields2Cover-targets
   LIBRARY DESTINATION ${INSTALL_LIB_DIR}
   ARCHIVE DESTINATION ${INSTALL_LIB_DIR}
@@ -265,9 +259,7 @@ install(
 export(
   TARGETS
     Fields2Cover
-    tinyxml2
     steering_functions
-    nlohmann_json
   FILE
     Fields2Cover-${type}-Targets.cmake
 )
