Use system-provided zenoh packages

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: zenoh-cpp-vendor-0.6.7-1/CMakeLists.txt
===================================================================
--- zenoh-cpp-vendor-0.6.7-1.orig/CMakeLists.txt
+++ zenoh-cpp-vendor-0.6.7-1/CMakeLists.txt
@@ -41,7 +41,10 @@ else()
     set(zenoh_c_commit b31348fa7f94f44f1f7b049c111a710e970a2725)
   endif()
 
+  find_package(zenohc)
+
   ament_vendor(zenoh_c_vendor
+    SATISFIED zenohc_FOUND
     VCS_URL https://github.com/eclipse-zenoh/zenoh-c.git
     VCS_VERSION ${zenoh_c_commit}
     CMAKE_ARGS
@@ -50,14 +53,16 @@ else()
       "-DZENOHC_CUSTOM_TARGET=${ZENOHC_CUSTOM_TARGET}"
   )
 
+  find_package(zenohcxx)
+
   ament_vendor(zenoh_cpp_vendor
+    SATISFIED zenohcxx_FOUND
     VCS_URL https://github.com/eclipse-zenoh/zenoh-cpp
     VCS_VERSION af381b420cc8837ac7da42c9984594ef8f110e90
     CMAKE_ARGS
       -DZENOHCXX_ZENOHC=OFF
   )
 
-  externalproject_add_stepdependencies(zenoh_cpp_vendor configure zenoh_c_vendor)
 endif()
 
 ament_export_dependencies(zenohc)
