From f4130241042b2c19dda7b6d3a22e822585b4901a Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmwcarit.de>
Date: Tue, 31 Dec 2013 16:20:04 +0100
Subject: [PATCH] image_view must explicitly require OpenCV's core component

---
 image_view/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/image_view/CMakeLists.txt b/image_view/CMakeLists.txt
index d72e992..e5b7790 100644
--- a/image_view/CMakeLists.txt
+++ b/image_view/CMakeLists.txt
@@ -6,7 +6,7 @@ catkin_package()
 
 find_package(Boost REQUIRED COMPONENTS signals thread)
 find_package(GTK2)
-find_package(OpenCV REQUIRED highgui)
+find_package(OpenCV REQUIRED core highgui)
 find_package(catkin REQUIRED camera_calibration_parsers cv_bridge image_transport message_filters nodelet rosconsole roscpp)
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS}
                            ${catkin_INCLUDE_DIRS}
-- 
1.8.1.4

