From f5dd2fc17dd7f634569b99141ce37db59975c248 Mon Sep 17 00:00:00 2001
From: AAlon <avishayalon@gmail.com>
Date: Sat, 7 Sep 2019 15:56:12 -0700
Subject: [PATCH] Guard test targets with if(BUILD_TESTING)

Upstream-Status: Pending

---
 CMakeLists.txt | 45 ++++++++++++++++++++++++---------------------
 1 file changed, 24 insertions(+), 21 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86a5d53..172262c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,6 @@ add_compile_options(-std=c++14)
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(aws_common REQUIRED)
-find_package(ament_cmake_gtest REQUIRED)
 
 if(AWSSDK_FOUND)
   set(SERVICE core)
@@ -33,26 +32,30 @@ install(
     DESTINATION include/
 )
 
-ament_add_gtest(test_configuration_provider test/client_configuration_provider_test.cpp)
-target_link_libraries(test_configuration_provider
-    ${aws_common_LIBRARIES}
-    ${PROJECT_NAME}
-    ${rclcpp_LIBRARIES}
-)
-
-ament_add_gtest(test_logger test/sdk_utils/logging/aws_ros_logger_test.cpp)
-target_link_libraries(test_logger
-    ${aws_common_LIBRARIES}
-    ${PROJECT_NAME}
-    ${rclcpp_LIBRARIES}
-)
-
-ament_add_gtest(test_parameter_reader test/parameter_reader_test.cpp)
-target_link_libraries(test_parameter_reader
-    ${aws_common_LIBRARIES}
-    ${PROJECT_NAME}
-    ${rclcpp_LIBRARIES}
-)
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
+  
+  ament_add_gtest(test_configuration_provider test/client_configuration_provider_test.cpp)
+  target_link_libraries(test_configuration_provider
+      ${aws_common_LIBRARIES}
+      ${PROJECT_NAME}
+      ${rclcpp_LIBRARIES}
+  )
+
+  ament_add_gtest(test_logger test/sdk_utils/logging/aws_ros_logger_test.cpp)
+  target_link_libraries(test_logger
+      ${aws_common_LIBRARIES}
+      ${PROJECT_NAME}
+      ${rclcpp_LIBRARIES}
+  )
+
+  ament_add_gtest(test_parameter_reader test/parameter_reader_test.cpp)
+  target_link_libraries(test_parameter_reader
+      ${aws_common_LIBRARIES}
+      ${PROJECT_NAME}
+      ${rclcpp_LIBRARIES}
+  )
+endif()
 
 ament_export_libraries(${PROJECT_NAME})
 ament_export_include_directories("include")
