From f95790326dde62e98894a61528f5374f15a10dcf Mon Sep 17 00:00:00 2001
From: Ismo Puustinen <ismo.puustinen@intel.com>
Date: Mon, 27 Feb 2017 13:16:23 +0200
Subject: [PATCH] build: always disable platform guessing.

Upstream-Status: Inappropriate

Signed-off-by: Ismo Puustinen <ismo.puustinen@intel.com>
---
 setup.py | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/setup.py b/setup.py
index 1dc146d..dcdb8e8 100755
--- a/setup.py
+++ b/setup.py
@@ -251,7 +251,7 @@ class pil_build_ext(build_ext):
         #
         # add platform directories
 
-        if self.disable_platform_guessing:
+        if self.disable_platform_guessing or True:
             pass
 
         elif sys.platform == "cygwin":
@@ -373,16 +365,6 @@ class pil_build_ext(build_ext):
 
         # FIXME: check /opt/stuff directories here?
 
-        # standard locations
-        if not self.disable_platform_guessing:
-            _add_directory(library_dirs, "/usr/local/lib")
-            _add_directory(include_dirs, "/usr/local/include")
-
-            _add_directory(library_dirs, "/usr/lib")
-            _add_directory(include_dirs, "/usr/include")
-            # alpine, at least
-            _add_directory(library_dirs, "/lib")
-
         # on Windows, look for the OpenJPEG libraries in the location that
         # the official installer puts them
         if sys.platform == "win32":
-- 
2.9.3

