From 0f19fb544663933680c4bfa110a63311c634f149 Mon Sep 17 00:00:00 2001
From: Jan Vermaete <jan.vermaete@gmail.com>
Date: Mon, 10 Aug 2026 19:26:52 +0200
Subject: [PATCH 2/2] ptest: correct path of the to be checked python files

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Jan Vermaete <jan.vermaete@gmail.com>
---
 test/test_flake8.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/test_flake8.py b/test/test_flake8.py
index de47084..bb48f88 100644
--- a/test/test_flake8.py
+++ b/test/test_flake8.py
@@ -3,6 +3,7 @@
 
 import logging
 from pathlib import Path
+import site
 import sys
 
 import pytest
@@ -33,7 +34,7 @@ def test_flake8():
     sys.stdout = sys.stderr
     # implicitly calls report_errors()
     report = style_guide.check_files([
-        str(Path(__file__).parents[1] / 'colcon_defaults'),
+        site.getsitepackages()[0] + '/' + 'colcon_defaults',
     ])
     report_tests = style_guide_tests.check_files([
         str(Path(__file__).parents[1] / 'test'),
-- 
2.43.0

