From e125886b0c497f01541868dc184130e33089f116 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Tue, 2 May 2017 08:42:14 +0200
Subject: [PATCH] also build with CATKIN_ENABLE_TESTING=OFF

Upstream-Status: Pending

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 474363f..20367d9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,13 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(urdf_tutorial)
 
-find_package(catkin REQUIRED roslaunch)
+find_package(catkin REQUIRED)
 catkin_package()
 
-roslaunch_add_file_check(launch)
+if (CATKIN_ENABLE_TESTING)
+    find_package(roslaunch REQUIRED)
+    roslaunch_add_file_check(launch)
+endif()
 
 install(DIRECTORY config images meshes launch rviz urdf
   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
-- 
1.9.1

