From f1a707824ae5a127f2e6f653523ebc00ba3c3041 Mon Sep 17 00:00:00 2001
From: JeongBong Seo <jb.seo@lge.com>
Date: Tue, 9 Apr 2019 13:13:18 +0900
Subject: Add yaml-cpp-vendor as dependency

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08310d3..5bfec7c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ find_package(camera_info_manager REQUIRED)
 find_package(image_transport REQUIRED)
 find_package(class_loader REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(yaml_cpp_vendor REQUIRED)
 
 set(opencv_2_components core highgui)
 set(opencv_3_components core imgcodecs videoio)
@@ -38,6 +39,7 @@ ament_target_dependencies(image_publisher_node
   "class_loader"
   "camera_info_manager"
   "rclcpp"
+  "yaml_cpp_vendor"
 )
 
 rclcpp_register_node_plugins(image_publisher_node "${PROJECT_NAME}::ImagePublisherNode")
-- 
2.17.1

