Index: git/src/georeference.cpp
===================================================================
--- git.orig/src/georeference.cpp
+++ git/src/georeference.cpp
@@ -108,10 +108,10 @@ namespace swri_transform_util
       // If we have an absolute path, we don't need to change it
       // Otherwise, we want to create a relative path from the .geo file
       // location
-      if (imagePath.is_complete() == false)
+      if (imagePath.is_absolute() == false)
       {
         boost::filesystem::path geoPath(path_);
-        image_path_ = (geoPath.parent_path() / imagePath.relative_path()).normalize().string();
+        image_path_ = (geoPath.parent_path() / imagePath.relative_path()).lexically_normal().string();
 
         RCLCPP_INFO(logger_, "georeference: Image path is %s", image_path_.c_str());
       }
