Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -9,14 +9,12 @@ find_package(orocos_kdl 1.5.1 QUIET)
 
 ament_vendor(orocos_kdl_vendor
   SATISFIED ${orocos_kdl_FOUND}
-  VCS_URL https://github.com/orocos/orocos_kinematics_dynamics.git
-  VCS_VERSION ce4bcb65a050615b6d7f21bc60fbb2656515791b
-  SOURCE_SUBDIR orocos_kdl
+  VCS_URL orocos-kdl/orocos_kdl
+  VCS_TYPE path
   GLOBAL_HOOK
   CMAKE_ARGS
     -DENABLE_TESTS:BOOL=OFF
     -DENABLE_EXAMPLES:BOOL=OFF
-  PATCHES patches
 )
 
 if(BUILD_TESTING)
