From: Jochen Sprickerhof <jochen@sprickerhof.de>
Date: Mon, 20 May 2013 13:06:29 +0200
Subject: Disable SSE

---
 ThirdParty/PSCommon/BuildSystem/Platform.x86 |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/ThirdParty/PSCommon/BuildSystem/Platform.x86
+++ b/ThirdParty/PSCommon/BuildSystem/Platform.x86
@@ -2,7 +2,7 @@
 export GLUT_SUPPORTED=1
 
 ifndef SSE_GENERATION
-	SSE_GENERATION = 3
+	SSE_GENERATION = 0
 endif
 
 ifeq ("$(OSTYPE)","Darwin")
@@ -19,7 +19,5 @@
 		ifeq ($(SSSE3_ENABLED), 1)
 			CFLAGS += -mssse3
 		endif
-	else
-		DUMMY:=($error "Only SSE2 and SSE3 are supported")
 	endif
 endif
