Remove Boost::system

Upstream-Status: Pending

Index: moveit-ros-planning-interface-2.5.9-1/ConfigExtras.cmake
===================================================================
--- moveit-ros-planning-interface-2.5.9-1.orig/ConfigExtras.cmake
+++ moveit-ros-planning-interface-2.5.9-1/ConfigExtras.cmake
@@ -11,6 +11,5 @@ find_package(Boost REQUIRED COMPONENTS
   filesystem
   program_options
   #${BOOST_PYTHON_COMPONENT}
-  system
   thread
 )
