From 4efc32314f5ee02e5be759ea288e11b6de64f1a1 Mon Sep 17 00:00:00 2001
From: Stefan Herbrechtsmeier <stefan@herbrechtsmeier.net>
Date: Tue, 4 Jun 2013 17:58:41 +0200
Subject: [PATCH] genlisp-extras.cmake.em: Use find_program for GENLISP_BIN

Signed-off-by: Stefan Herbrechtsmeier <stefan@herbrechtsmeier.net>
---
 cmake/genlisp-extras.cmake.em |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/cmake/genlisp-extras.cmake.em b/cmake/genlisp-extras.cmake.em
index 0d135d0..db52d54 100644
--- a/cmake/genlisp-extras.cmake.em
+++ b/cmake/genlisp-extras.cmake.em
@@ -1,13 +1,17 @@
 @[if DEVELSPACE]@
 # bin and template dir variables in develspace
-set(GENLISP_BIN "@(CMAKE_CURRENT_SOURCE_DIR)/scripts/gen_lisp.py")
+set(GENLISP_BIN_DIR "@(CMAKE_CURRENT_SOURCE_DIR)/scripts/gen_lisp.py")
 set(GENLISP_TEMPLATE_DIR "@(CMAKE_CURRENT_SOURCE_DIR)/scripts")
+set(CATKIN_FIND_ROOT_PATH_MODE NO_CMAKE_FIND_ROOT_PATH)
 @[else]@
 # bin and template dir variables in installspace
-set(GENLISP_BIN "@(CMAKE_INSTALL_PREFIX)/@(CATKIN_PACKAGE_BIN_DESTINATION)/gen_lisp.py")
-set(GENLISP_TEMPLATE_DIR "@(CMAKE_INSTALL_PREFIX)/@(CATKIN_PACKAGE_SHARE_DESTINATION)")
+set(GENLISP_BIN_DIR "@(CMAKE_INSTALL_PREFIX)/@(CATKIN_PACKAGE_BIN_DESTINATION)/")
+set(GENLISP_TEMPLATE_DIR "${CMAKE_CURRENT_LIST_DIR}/..")
+set(CATKIN_FIND_ROOT_PATH_MODE "")
 @[end if]@
 
+find_program(GENLISP_BIN gen_lisp.py PATHS ${GENLISP_BIN_DIR} NO_DEFAULT_PATH ${CATKIN_FIND_ROOT_PATH_MODE})
+
 # Generate .msg or .srv -> .lisp
 # The generated .lisp files should be added ALL_GEN_OUTPUT_FILES_lisp
 macro(_generate_lisp ARG_PKG ARG_MSG ARG_IFLAGS ARG_MSG_DEPS ARG_GEN_OUTPUT_DIR)
-- 
1.7.0.4

