CMakeLists.txt: Fetch uncrustify with bitbake fetcher

Upstream-Status: Not applicable

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,9 +30,9 @@ endif()
 
 ament_vendor(uncrustify_vendor
   SATISFIED ${UNCRUSTIFY}
-  VCS_URL https://github.com/uncrustify/uncrustify.git
+  VCS_TYPE path
+  VCS_URL ${CMAKE_SOURCE_DIR}/uncrustify-upstream
   VCS_VERSION uncrustify-0.72.0
-  PATCHES patches
   CMAKE_ARGS
     -DNoGitVersionString=ON
 )
