Allow newer LLVM versions

Upstream Status: Inappropriate

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

diff --git a/frontend/llvm/CMakeLists.txt b/frontend/llvm/CMakeLists.txt
index cef8fd1..76af5a9 100644
--- a/frontend/llvm/CMakeLists.txt
+++ b/frontend/llvm/CMakeLists.txt
@@ -103,10 +104,6 @@ include_directories(${AR_INCLUDE_DIR})
 find_package(LLVM REQUIRED)
 include_directories(SYSTEM ${LLVM_INCLUDE_DIR})
 
-if ((LLVM_VERSION VERSION_LESS "14") OR (NOT (LLVM_VERSION VERSION_LESS "15")))
-  message(FATAL_ERROR "LLVM 14 is required.")
-endif()
-
 # Add path to llvm cmake modules
 list(APPEND CMAKE_MODULE_PATH ${LLVM_CMAKE_DIR})
 
