From 9dfdfa5a861855449100f34c638ba494cf77c83b Mon Sep 17 00:00:00 2001
From: AAlon <avishaya@amazon.com>
Date: Sat, 7 Sep 2019 18:51:05 -0700
Subject: [PATCH] Guard test targets with if(BUILD_TESTING)

Upstream-Status: Backport [https://github.com/aws-robotics/kinesisvideo-encoder-ros2/commit/a7bef6d135ca85cce1ddfd65a9b776c3fda34ccb]
---
 CMakeLists.txt | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index de4e680..853bd94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,18 +4,15 @@ project(h264_video_encoder)
 ## Compile as C++14
 set(CMAKE_CXX_STANDARD 14)
 
-
 find_package(h264_encoder_core REQUIRED)
 find_package(aws_common REQUIRED)
 find_package(aws_ros2_common REQUIRED)
 find_package(image_transport REQUIRED)
 find_package(sensor_msgs REQUIRED)
 find_package(kinesis_video_msgs REQUIRED)
-find_package(ament_cmake_gtest REQUIRED)
 find_package(ament_cmake REQUIRED)
 find_package(rmw_implementation REQUIRED)
 
-
 #############
 ## Compile ##
 #############
@@ -74,24 +71,27 @@ install(DIRECTORY config DESTINATION share/${PROJECT_NAME})
 ## Tests ##
 #############
 
-## Add gtest based cpp test target and link libraries
-ament_add_gtest(test_h264_video_encoder
-  test/h264_video_encoder_test.cpp
-)
-target_include_directories(test_h264_video_encoder
-PRIVATE include ${h264_video_encoder_INCS})
-target_link_libraries(test_h264_video_encoder
-  ${PROJECT_NAME}_lib
-  ${h264_video_encoder_LIBS}
-  ${GMOCK_BOTH_LIBRARIES}
-)
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
+  ## Add gtest based cpp test target and link libraries
+  ament_add_gtest(test_h264_video_encoder
+    test/h264_video_encoder_test.cpp
+  )
+  target_include_directories(test_h264_video_encoder
+  PRIVATE include ${h264_video_encoder_INCS})
+  target_link_libraries(test_h264_video_encoder
+    ${PROJECT_NAME}_lib
+    ${h264_video_encoder_LIBS}
+    ${GMOCK_BOTH_LIBRARIES}
+  )
+endif()
+
 ament_export_dependencies(h264_encoder_core)
 ament_export_dependencies(aws_common)
 ament_export_dependencies(aws_ros2_common)
 ament_export_dependencies(image_transport)
 ament_export_dependencies(sensor_msgs)
 ament_export_dependencies(kinesis_video_msgs)
-ament_export_dependencies(ament_cmake_gtest)
 ament_export_dependencies(ament_cmake)
 ament_export_dependencies(rmw_implementation)
 
-- 
2.17.1

