Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -11,7 +11,10 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aar
   set(EXTRA_CMAKE_FLAGS -DOGRE_SIMD_NEON:BOOL=FALSE  -DOGRE_SIMD_SSE2:BOOL=FALSE)
 endif()

+find_package(OGRE-Next)
+
 ament_vendor(${PROJECT_NAME}
+  SATISFIED OGRE-Next_FOUND
   VCS_URL https://github.com/OGRECave/ogre-next.git
   VCS_VERSION v2.3.3
   CMAKE_ARGS
