From 466c37a950af6601a6b125e11703d8ee39fca303 Mon Sep 17 00:00:00 2001
From: Amelia Grace <agrace@bastiansolutions.com>
Date: Wed, 30 Apr 2025 14:05:54 -0600
Subject: [PATCH] remove missing braces error compile option

Signed-off-by: Amelia Grace <agrace@bastiansolutions.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5eedec32e6..c730b0a82d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic -Werror=conversion -Werror=unused-but-set-variable
-  -Werror=return-type -Werror=format -Werror=missing-braces)
+  -Werror=return-type -Werror=format)
 endif()
 
 # find dependencies
