From 0e9f364cad9082daac0d802abc690b1f59ac0243 Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Tue, 11 Jul 2017 12:53:38 +0300
Subject: [PATCH] get rid of poco-vendor dependency

Upstream-Status: Inappropriate [poco is provided by OE]
---
 rmw_implementation/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/rmw_implementation/CMakeLists.txt b/rmw_implementation/CMakeLists.txt
index c23861e..e9ed54f 100644
--- a/rmw_implementation/CMakeLists.txt
+++ b/rmw_implementation/CMakeLists.txt
@@ -16,7 +16,6 @@ find_package(ament_cmake REQUIRED)
 find_package(rcutils REQUIRED)
 
 # provides FindPoco.cmake and Poco on platforms without it
-find_package(poco_vendor)
 find_package(Poco COMPONENTS Foundation)
 find_package(rmw_implementation_cmake REQUIRED)
 
-- 
2.9.3

