From 289cf84550fe47d9528f3e1b1d7f62d802e3268c Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 21 May 2020 10:14:23 +0000
Subject: [PATCH] common/fw/CMakeLists.txt: respect *_FW_URL when set

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 common/fw/CMakeLists.txt | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/common/fw/CMakeLists.txt b/common/fw/CMakeLists.txt
index 94d9fc9..e5a9e11 100644
--- a/common/fw/CMakeLists.txt
+++ b/common/fw/CMakeLists.txt
@@ -15,26 +15,34 @@ string(REGEX MATCH "D4XX_RC_FIRMWARE_VERSION \"([0-9]+.[0-9]+.[0-9]+.[0-9]+)\""
 set(D4XX_RC_VERSION ${CMAKE_MATCH_1})
 message(STATUS "D4XX_RC_VERSION: ${D4XX_RC_VERSION}")
 set(D4XX_RC_SHA1 be77db800142110b35d1cc7f12dfc77e75ed9c8a)
-set(D4XX_RC_URL "${REALSENSE_FIRMWARE_URL}/Releases/RS4xx/FW")
+if (NOT DEFINED D4XX_RC_URL)
+  set(D4XX_RC_URL "${REALSENSE_FIRMWARE_URL}/Releases/RS4xx/FW")
+endif()
 
 string(REGEX MATCH "D4XX_RECOMMENDED_FIRMWARE_VERSION \"([0-9]+.[0-9]+.[0-9]+.[0-9]+)\"" _ ${ver})
 set(D4XX_FW_VERSION ${CMAKE_MATCH_1})
 message(STATUS "D4XX_FW_VERSION: ${D4XX_FW_VERSION}")
 set(D4XX_FW_SHA1 be77db800142110b35d1cc7f12dfc77e75ed9c8a)
-set(D4XX_FW_URL "${REALSENSE_FIRMWARE_URL}/Releases/RS4xx/FW")
+if (NOT DEFINED D4XX_FW_URL)
+  set(D4XX_FW_URL "${REALSENSE_FIRMWARE_URL}/Releases/RS4xx/FW")
+endif()
 
 
 string(REGEX MATCH "SR3XX_RECOMMENDED_FIRMWARE_VERSION \"([0-9]+.[0-9]+.[0-9]+.[0-9]+)\"" _ ${ver})
 set(SR3XX_FW_VERSION ${CMAKE_MATCH_1})
 message(STATUS "SR3XX_FW_VERSION: ${SR3XX_FW_VERSION}")
 set(SR3XX_FW_SHA1 55237dba5d7db20e7c218975375d05b4210e9460)
-set(SR3XX_FW_URL "${REALSENSE_FIRMWARE_URL}/Releases/SR300/FW")
+if (NOT DEFINED SR3XX_FW_URL)
+  set(SR3XX_FW_URL "${REALSENSE_FIRMWARE_URL}/Releases/SR300/FW")
+endif()
 
 string(REGEX MATCH "T26X_FIRMWARE_VERSION \"([0-9]+.[0-9]+.[0-9]+.[0-9]+)\"" _ ${ver})
 set(T26X_FW_VERSION ${CMAKE_MATCH_1})
 message(STATUS "T26X_FW_VERSION: ${T26X_FW_VERSION}")
 set(T26X_FW_SHA1 fb857641b003977478c4e5855658803e13a784c7)
-set(T26X_FW_URL "${REALSENSE_FIRMWARE_URL}/Releases/TM2/FW/target/${T26X_FW_VERSION}")
+if (NOT DEFINED T26X_FW_URL)
+  set(T26X_FW_URL "${REALSENSE_FIRMWARE_URL}/Releases/TM2/FW/target/${T26X_FW_VERSION}")
+endif()
 
 add_library(${PROJECT_NAME} STATIC empty.c)
 
