diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8e73ba..ffe48ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,6 +52,6 @@ install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})
 
 install(TARGETS ${PROJECT_NAME}
   EXPORT export_${PROJECT_NAME}
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION bin)
