From f354a7264519cbb40473e0de6539b622ddb0579c Mon Sep 17 00:00:00 2001
From: Herb Kuta <herb.kuta@lge.com>
Date: Tue, 2 Apr 2019 16:30:36 -0700

Upstream-Status: Pending

---
 Makefile | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/Makefile b/Makefile
index a72cc6c62..bc5a603ff 100644
--- a/Makefile
+++ b/Makefile
@@ -102,9 +102,11 @@ ifeq ($(HAVE_RULES),yes)
     endif
 endif
 
-ifndef PREFIX
-    PREFIX=/usr
-endif
+DESTDIR?= ""
+prefix?= "/usr"
+bindir?= "$(prefix)/bin"
+BINDIR=$(bindir)
+PREFIX=$(prefix)
 
 ifndef INCLUDE_FOR_LIB
     INCLUDE_FOR_LIB=-Ilib -Iexternals/simplecpp -Iexternals/tinyxml
@@ -118,8 +120,6 @@ ifndef INCLUDE_FOR_TEST
     INCLUDE_FOR_TEST=-Ilib -Icli -Iexternals/simplecpp -Iexternals/tinyxml
 endif
 
-BIN=$(DESTDIR)$(PREFIX)/bin
-
 # For 'make man': sudo apt-get install xsltproc docbook-xsl docbook-xml on Linux
 DB2MAN?=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
 XP=xsltproc -''-nonet -''-param man.charmap.use.subset "0"
@@ -292,11 +292,11 @@ tags:
 	ctags -R --exclude=doxyoutput --exclude=test/cfg --exclude=test/synthetic cli externals gui lib test
 
 install: cppcheck
-	install -d ${BIN}
-	install cppcheck ${BIN}
-	install addons/*.py ${BIN}
-	install addons/*/*.py ${BIN}
-	install htmlreport/cppcheck-htmlreport ${BIN}
+	install -d ${DESTDIR}${BINDIR}
+	install cppcheck ${DESTDIR}${BINDIR}
+	install addons/*.py ${DESTDIR}${BINDIR}
+	install addons/*/*.py ${DESTDIR}${BINDIR}
+	install htmlreport/cppcheck-htmlreport ${DESTDIR}${BINDIR}
 ifdef CFGDIR 
 	install -d ${DESTDIR}${CFGDIR}
 	install -m 644 cfg/* ${DESTDIR}${CFGDIR}
