From b95911a0a52fd7f2e0b0f9cf1bb15fbf834bd5d1 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 2 Nov 2020 14:24:31 -0800
Subject: [PATCH] Revert: cmake: fix project packaging

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3e7e17..26b3075 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -175,8 +175,7 @@ TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME}
   ${EIGEN3_INCLUDE_DIR}
   ${PYTHON_INCLUDE_DIRS} 
   ${NUMPY_INCLUDE_DIRS}
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
-  $<INSTALL_INTERFACE:include>)
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
 IF(SUFFIX_SO_VERSION)
   SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION})
@@ -200,6 +199,9 @@ INSTALL(TARGETS ${PROJECT_NAME}
 ADD_HEADER_GROUP(${PROJECT_NAME}_HEADERS)
 ADD_SOURCE_GROUP(${PROJECT_NAME}_SOURCES)
 
+TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME} INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>")
+SET(PUBLIC_HEADER ${${PROJECT_NAME}_HEADERS})
+
 # Install package for ROS
 install(FILES package.xml DESTINATION share/eigenpy)
 
