Disable CMake FetchContent and use Bitbake fetcher instead.

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddf42b7..7eeb8a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,13 +1,4 @@
 cmake_minimum_required(VERSION 3.14)
-include(FetchContent)
-#include(ExternalProject)
-FetchContent_Declare(
-  kinova_binary_api
-  URL https://artifactory.kinovaapps.com:443/artifactory/generic-public/kortex/API/2.5.0/linux_x86-64_x86_gcc.zip
-  URL_HASH MD5=64bd86e7ab8bda90ef1fc7d6a356e080
-)
-
-FetchContent_MakeAvailable(kinova_binary_api)
 
 project(kortex_api)
 
@@ -26,14 +17,8 @@ else()
   message(FATAL_ERROR "Unsupported System: currently support is for Linux x68_64. Detected ${CMAKE_SYSTEM_NAME} and ${CMAKE_SYSTEM_PROCESSOR}")
 endif()
 
-#ExternalProject_Add(kinova_binary_api
-#  URL ${API_URL}
-#  CONFIGURE_COMMAND ""
-#  BUILD_COMMAND ""
-#  INSTALL_COMMAND ""
-#)
-
 find_package(ament_cmake REQUIRED)
+set(kinova_binary_api_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/kinova_binary_api")
 
 ament_export_include_directories(include/kortex_api)
 
