Upstream-Status: Inactive-Upstream [lastrelease: Jan 27, 2010]

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

From: Teemu Ikonen <tpikonen@gmail.com>
Date: Mon, 26 Mar 2018 12:39:44 +0300
Subject: [PATCH] Do not remove figures on doc/Makefile clean target.

---
 doc/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/Makefile b/doc/Makefile
index 4e52908..d64cb49 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -77,7 +77,7 @@ include ../Make-config
 #-----------------------------------------------------------------------------
 
 clean:
-	-rm -f core *.aux *.dvi *.log *.blg *.bbl *.toc $(FIGURES);
+	-rm -f core *.aux *.dvi *.log *.blg *.bbl *.toc # $(FIGURES);
 
 realclean: clean
 	-rm -f $(ANNMAN).ps
