From 1dee61dabbf3b5de2170ca15ecea3ed93d57339d Mon Sep 17 00:00:00 2001
From: vmayoral <v.mayoralv@gmail.com>
Date: Sat, 17 Aug 2013 17:40:20 +0200
Subject: [PATCH] joint-limits-interface CMakeLists fix

---
 joint_limits_interface/CMakeLists.txt |   16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/joint_limits_interface/CMakeLists.txt b/joint_limits_interface/CMakeLists.txt
index 6e18253..1561b22 100644
--- a/joint_limits_interface/CMakeLists.txt
+++ b/joint_limits_interface/CMakeLists.txt
@@ -43,16 +43,18 @@ else()
       ${urdfdom_LIBRARIES}
   )
 
-  catkin_add_gtest(joint_limits_interface_test test/joint_limits_interface_test.cpp)
-  target_link_libraries(joint_limits_interface_test ${catkin_LIBRARIES})
+  if(CATKIN_ENABLE_TESTING)
+    catkin_add_gtest(joint_limits_interface_test test/joint_limits_interface_test.cpp)
+    target_link_libraries(joint_limits_interface_test ${catkin_LIBRARIES})
 
-  catkin_add_gtest(joint_limits_urdf_test      test/joint_limits_urdf_test.cpp)
-  target_link_libraries(joint_limits_urdf_test ${catkin_LIBRARIES})
+    catkin_add_gtest(joint_limits_urdf_test      test/joint_limits_urdf_test.cpp)
+    target_link_libraries(joint_limits_urdf_test ${catkin_LIBRARIES})
 
-  catkin_add_gtest(joint_limits_rosparam_test  test/joint_limits_urdf_test.cpp)
-  target_link_libraries(joint_limits_rosparam_test ${catkin_LIBRARIES})
+    catkin_add_gtest(joint_limits_rosparam_test  test/joint_limits_urdf_test.cpp)
+    target_link_libraries(joint_limits_rosparam_test ${catkin_LIBRARIES})
 
-  add_rostest(test/joint_limits_rosparam.test)
+    add_rostest(test/joint_limits_rosparam.test)
+  endif()
 
   # Install
   install(DIRECTORY include/${PROJECT_NAME}/
-- 
1.7.9.5

