From 9f1e801c6de00287658f1bae21048f37bb9bb198 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Sat, 21 Sep 2013 16:13:16 +0200
Subject: [PATCH] rostest is optional

---
 tf/CMakeLists.txt |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tf/CMakeLists.txt b/tf/CMakeLists.txt
index e8d3044..2d36df3 100644
--- a/tf/CMakeLists.txt
+++ b/tf/CMakeLists.txt
@@ -4,7 +4,12 @@ project(tf)
 find_package(catkin REQUIRED)
 
 find_package(Boost REQUIRED thread signals)
-find_package(catkin REQUIRED angles geometry_msgs message_filters message_generation rosconsole roscpp rostest rostime sensor_msgs std_msgs tf2_ros)
+
+if(CATKIN_ENABLE_TESTING)
+  find_package(catkin REQUIRED angles geometry_msgs message_filters message_generation rosconsole roscpp rostest rostime sensor_msgs std_msgs tf2_ros)
+else()
+  find_package(catkin REQUIRED angles geometry_msgs message_filters message_generation rosconsole roscpp rostime sensor_msgs std_msgs tf2_ros)
+endif()
 
 catkin_python_setup()
 
-- 
1.7.10.4

