From 18e81e33c052e26ae60259708095a2a933f75fd1 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Tue, 4 Feb 2014 16:02:48 +0100
Subject: [PATCH] make rostest in CMakeLists optional (ros/rosdistro#3010)

Upstream-Status: Backport [Accepted in indigo-devel branch]
---
 tf/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tf/CMakeLists.txt b/tf/CMakeLists.txt
index fa4a42f..c6988e1 100644
--- a/tf/CMakeLists.txt
+++ b/tf/CMakeLists.txt
@@ -4,7 +4,7 @@ project(tf)
 find_package(catkin REQUIRED)
 
 find_package(Boost REQUIRED thread signals)
-find_package(catkin REQUIRED angles geometry_msgs message_filters message_generation rosconsole roscpp rostest rostime sensor_msgs std_msgs tf2_ros)
+find_package(catkin REQUIRED angles geometry_msgs message_filters message_generation rosconsole roscpp rostime sensor_msgs std_msgs tf2_ros)
 
 catkin_python_setup()
 
@@ -48,6 +48,8 @@ target_link_libraries(static_transform_publisher ${PROJECT_NAME})
 # Tests
 if(CATKIN_ENABLE_TESTING)
 
+find_package(rostest)
+
 catkin_add_gtest(tf_unittest test/tf_unittest.cpp)
 target_link_libraries(tf_unittest ${PROJECT_NAME})
 
-- 
1.8.3.2

