From f6ae8b45b89640c5a9a69dfaa8e25ab3798be686 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 5 Nov 2020 15:17:12 -0800
Subject: [PATCH] CMakeLists.txt: fetch osqp with bitbake fetcher

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddeb7bf..6d4bd9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,10 +14,7 @@ macro(build_osqp)
 
   include(ExternalProject)
   externalproject_add(osqp-v0.6.1.dev0
-    GIT_REPOSITORY https://github.com/oxfordcontrol/osqp.git
-    GIT_TAG v0.6.1.dev0
-    GIT_SHALLOW ON
-    TIMEOUT 60
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/osqp-upstream
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/osqp_install
       ${extra_cmake_args}
