Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -3,35 +3,54 @@ project(tinyspline_vendor)
 
 find_package(ament_cmake REQUIRED)
 
-include(FetchContent)
-fetchcontent_declare(
-  tinyspline
-  GIT_REPOSITORY https://github.com/msteinbeck/tinyspline.git
-  GIT_TAG        5f0e8fad34e86771cfae3dac4beef5f1858d5610 # v0.6.0
-)
 set(BUILD_SHARED_LIBS ON)
-fetchcontent_makeavailable(tinyspline)
+set(tinyspline_SOURCE_DIR ${CMAKE_SOURCE_DIR}/tinyspline)
+set(tinyspline_BINARY_DIR ${CMAKE_BINARY_DIR}/tinyspline-install)
+
+include(ExternalProject)
+externalproject_add(tinyspline
+    ${cmake_commands}
+    SOURCE_DIR ${PROJECT_SOURCE_DIR}/tinyspline
+    CMAKE_ARGS
+      -DCMAKE_INSTALL_PREFIX=${tinyspline_BINARY_DIR}
+      ${cmake_configure_args}
+      -DTINYSPLINE_ENABLE_LUA=True
+      -DTINYSPLINE_ENABLE_PYTHON=True
+      -DLUA_SOURCE_DIR=${LUA_SOURCE_DIR}
+)
 
 install(
   DIRECTORY
-    ${tinyspline_BINARY_DIR}/src/CMakeFiles/Export/lib64/cmake/tinyspline/
+    ${tinyspline_BINARY_DIR}/lib64/cmake/tinyspline/
   DESTINATION
     share/${PROJECT_NAME}/cmake
 )
 install(
   DIRECTORY
-    ${tinyspline_BINARY_DIR}/src/CMakeFiles/Export/lib64/cmake/tinysplinecxx/
+    ${tinyspline_BINARY_DIR}/lib64/cmake/tinysplinecxx/
   DESTINATION
     share/${PROJECT_NAME}/cmake
 )
 install(
   FILES
-    ${tinyspline_BINARY_DIR}/tinyspline-config.cmake
-    ${tinyspline_BINARY_DIR}/tinyspline-config-version.cmake
-    ${tinyspline_BINARY_DIR}/tinysplinecxx-config.cmake
-    ${tinyspline_BINARY_DIR}/tinysplinecxx-config-version.cmake
+    ${tinyspline_BINARY_DIR}/include/tinyspline.h
+    ${tinyspline_BINARY_DIR}/include/tinysplinecxx.h
   DESTINATION
-    share/${PROJECT_NAME}/cmake
+    ${CMAKE_INSTALL_INCLUDEDIR}
+)
+install(
+  FILES
+    ${tinyspline_BINARY_DIR}/lib64/libtinyspline.a
+    ${tinyspline_BINARY_DIR}/lib64/libtinysplinecxx.a
+  DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}
+)
+install(
+  FILES
+    ${tinyspline_BINARY_DIR}/lib64/pkgconfig/tinyspline.pc
+    ${tinyspline_BINARY_DIR}/lib64/pkgconfig/tinysplinecxx.pc
+  DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
 
 if(BUILD_TESTING)
