diff -uNr cartographer_ros-release-release-melodic-cartographer_ros-1.0.0-1/CMakeLists.txt cartographer_ros-release-release-melodic-cartographer_ros-1.0.0-1.new/CMakeLists.txt
--- cartographer_ros-release-release-melodic-cartographer_ros-1.0.0-1/CMakeLists.txt	2018-06-08 02:10:43.000000000 -0700
+++ cartographer_ros-release-release-melodic-cartographer_ros-1.0.0-1.new/CMakeLists.txt	2019-09-07 15:01:55.262919297 -0700
@@ -99,10 +99,10 @@
 target_link_libraries(${PROJECT_NAME} PUBLIC cartographer)
 
 # Lua
-target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${LUA_INCLUDE_DIR})
+target_include_directories(${PROJECT_NAME} PUBLIC ${LUA_INCLUDE_DIR})
 
 # PCL
-target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${PCL_INCLUDE_DIRS})
+target_include_directories(${PROJECT_NAME} PUBLIC ${PCL_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME} PUBLIC ${PCL_LIBRARIES})
 set(BLACKLISTED_PCL_DEFINITIONS " -march=native -msse4.2 -mfpmath=sse ")
 foreach(DEFINITION ${PCL_DEFINITIONS})
@@ -114,16 +114,16 @@
 endforeach()
 
 # Eigen
-target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
+target_include_directories(${PROJECT_NAME} PUBLIC
   "${EIGEN3_INCLUDE_DIR}")
 
 # Boost
-target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
+target_include_directories(${PROJECT_NAME} PUBLIC
   "${Boost_INCLUDE_DIRS}")
 target_link_libraries(${PROJECT_NAME} PUBLIC ${Boost_LIBRARIES})
 
 # Catkin
-target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${catkin_INCLUDE_DIRS})
+target_include_directories(${PROJECT_NAME} PUBLIC ${catkin_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME} PUBLIC ${catkin_LIBRARIES})
 add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
 
@@ -145,9 +145,9 @@
     catkin_add_gtest(${TEST_NAME} ${TEST_SOURCE_FILENAME})
     # catkin_add_gtest uses a plain (i.e. no PUBLIC/PRIVATE/INTERFACE) call to
     # target_link_libraries. That forces us to do the same.
-    target_include_directories(${TEST_NAME} SYSTEM PUBLIC ${LUA_INCLUDE_DIR})
+    target_include_directories(${TEST_NAME} PUBLIC ${LUA_INCLUDE_DIR})
     target_link_libraries(${TEST_NAME} ${LUA_LIBRARIES})
-    target_include_directories(${TEST_NAME} SYSTEM PUBLIC ${catkin_INCLUDE_DIRS})
+    target_include_directories(${TEST_NAME} PUBLIC ${catkin_INCLUDE_DIRS})
     target_link_libraries(${TEST_NAME} ${catkin_LIBRARIES})
     add_dependencies(${TEST_NAME} ${catkin_EXPORTED_TARGETS})
     target_link_libraries(${TEST_NAME} cartographer)
