Index: git/src/data.cpp
===================================================================
--- git.orig/src/data.cpp
+++ git/src/data.cpp
@@ -1,3 +1,5 @@
+#include <string>
+
 #include "create/data.h"
 
 #define ADD_PACKET(id,nbytes,info,enabledVersion) if ((enabledVersion) & version) packets[id]=std::make_shared<Packet>(nbytes,info)
Index: git/src/packet.cpp
===================================================================
--- git.orig/src/packet.cpp
+++ git/src/packet.cpp
@@ -1,3 +1,4 @@
+#include <string>
 #include <memory>
 
 #include "create/packet.h"
