From 597828ebc85f552fd38537664cbe72c1c775049b Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Sun, 13 Oct 2013 05:41:53 +0200
Subject: [PATCH] check for CATKIN_ENABLE_TESTING

---
 CMakeLists.txt | 6 ++++--
 package.xml    | 2 +-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b41961..821f929 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,8 +64,10 @@ add_library(${PROJECT_NAME}
 target_link_libraries(${PROJECT_NAME} ${ASSIMP_LIBRARIES} ${QHULL_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
 
 
-# Unit tests
-add_subdirectory(test)
+if(CATKIN_ENABLE_TESTING)
+  # Unit tests
+  add_subdirectory(test)
+endif()
 
 install(TARGETS ${PROJECT_NAME}
         ARCHIVE DESTINATION lib
diff --git a/package.xml b/package.xml
index 4001019..4dbd6d1 100644
--- a/package.xml
+++ b/package.xml
@@ -8,7 +8,7 @@
   <license>BSD</license>  
   <url>http://ros.org/wiki/geometric_shapes</url>
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend version_gte="0.5.68">catkin</buildtool_depend>
 
   <build_depend>boost</build_depend>
   <build_depend>shape_msgs</build_depend>
-- 
1.8.1.2

