Index: git/cmake/ros2_control.cmake
===================================================================
--- git.orig/cmake/ros2_control.cmake
+++ git/cmake/ros2_control.cmake
@@ -34,11 +34,6 @@ endmacro()
 # set compiler options depending on detected compiler
 macro(set_compiler_options)
   if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
-    add_compile_options(-Wall -Wextra -Wpedantic -Werror=conversion -Werror=unused-but-set-variable
-                        -Werror=return-type -Werror=shadow -Werror=format
-                        -Werror=missing-braces)
-    message(STATUS "Compiler warnings enabled for ${CMAKE_CXX_COMPILER_ID}")
-
     # https://docs.ros.org/en/rolling/How-To-Guides/Ament-CMake-Documentation.html#compiler-and-linker-options
     if(NOT CMAKE_C_STANDARD)
       set(CMAKE_C_STANDARD 99)
