From f2769df973af39e56538c70ebf2950929135f777 Mon Sep 17 00:00:00 2001
From: Eugene Golushkov <e.golushkov@nospam-gmail.com>
Date: Mon, 15 Nov 2021 19:41:27 +0200
Subject: [PATCH] Neon is architectural for AArch64

---
 OgreMain/include/OgrePlatform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/OgreMain/include/OgrePlatform.h b/OgreMain/include/OgrePlatform.h
index 8a9826bd80..c94904aebe 100644
--- a/OgreMain/include/OgrePlatformInformation.h
+++ b/OgreMain/include/OgrePlatformInformation.h
@@ -101,7 +101,7 @@ 
 
     /* Define whether or not Ogre compiled with NEON support.
      */
-    #if OGRE_DOUBLE_PRECISION == 0 && OGRE_CPU == OGRE_CPU_ARM && (OGRE_COMPILER == OGRE_COMPILER_GNUC || OGRE_COMPILER == OGRE_COMPILER_CLANG) && defined(__ARM_NEON__)
+    #if OGRE_DOUBLE_PRECISION == 0 && OGRE_CPU == OGRE_CPU_ARM && ( defined(__aarch64__) || defined(__ARM_NEON__) )
     #   define __OGRE_HAVE_NEON  1
     #endif
 #endif
-- 
2.43.0

